<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Network Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="00_all_files/libs/clipboard/clipboard.min.js"></script>
<script src="00_all_files/libs/quarto-html/quarto.js"></script>
<script src="00_all_files/libs/quarto-html/popper.min.js"></script>
<script src="00_all_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="00_all_files/libs/quarto-html/anchor.min.js"></script>
<link href="00_all_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="00_all_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="00_all_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="00_all_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="00_all_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Network Analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="load-data" class="level1">
<h1>Load Data</h1>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-raw-data" class="level2">
<h2 class="anchored" data-anchor-id="load-raw-data">Load Raw Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">"../data/_raw"</span>)) {</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">"../data/_raw"</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>raw_file <span class="ot">&lt;-</span> <span class="st">"../data/_raw/GSE45827.RData"</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(raw_file)) {</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  gse <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE45827"</span>, <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">save</span>(gse, <span class="at">file =</span> raw_file)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(raw_file)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="split-raw-data-into-3-main-data-frames" class="level2">
<h2 class="anchored" data-anchor-id="split-raw-data-into-3-main-data-frames">Split Raw Data into 3 main Data Frames</h2>
<ul>
<li><p>Gene expression Data</p></li>
<li><p>Sample Metadata</p></li>
<li><p>Gene Metadata</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>exprSet <span class="ot">&lt;-</span> gse[[<span class="dv">1</span>]]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">exprs</span>(exprSet)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">pData</span>(exprSet)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> <span class="fu">fData</span>(exprSet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="save-data-frames-into-csv-files" class="level2">
<h2 class="anchored" data-anchor-id="save-data-frames-into-csv-files">Save Data Frames into csv files</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(expr_data, <span class="st">"../data/GSE45827_expression.csv"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(pheno_data, <span class="st">"../data/GSE45827_Phenotype_metadata.csv"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(feature_data, <span class="st">"../data/GSE45827_Feature_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="clean-data" class="level1">
<h1>Clean Data</h1>
<section id="load-libraries-1" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-1">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data-1" class="level2">
<h2 class="anchored" data-anchor-id="load-data-1">Load Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_expression.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_Feature_metadata.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_Phenotype_metadata.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="phenotype-data-cleaning" class="level2">
<h2 class="anchored" data-anchor-id="phenotype-data-cleaning">Phenotype Data Cleaning</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>cell_lines <span class="ot">&lt;-</span> pheno_data <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="st">`</span><span class="at">source_name_ch1</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"Human CellLine"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> pheno_data <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="st">`</span><span class="at">source_name_ch1</span><span class="st">`</span> <span class="sc">!=</span> <span class="st">"Human CellLine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> pheno_data <span class="sc">|&gt;</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="st">"diagnosis.ch1"</span>, <span class="st">"tumor.subtype.ch1"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="st">`</span><span class="at">diagnosis.ch1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">if_else</span>(<span class="st">`</span><span class="at">diagnosis.ch1</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"None (normal)"</span>, <span class="st">"Healthy"</span>, <span class="st">`</span><span class="at">diagnosis.ch1</span><span class="st">`</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="st">`</span><span class="at">tumor.subtype.ch1</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">if_else</span>(<span class="st">`</span><span class="at">tumor.subtype.ch1</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"N/A"</span>, <span class="st">"Healthy"</span>,<span class="st">`</span><span class="at">tumor.subtype.ch1</span><span class="st">`</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">rename</span>(<span class="st">"diagnosis"</span> <span class="ot">=</span><span class="st">"diagnosis.ch1"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"subtype"</span> <span class="ot">=</span> <span class="st">"tumor.subtype.ch1"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">diagnosis =</span> <span class="fu">factor</span>(diagnosis),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">subtype =</span> <span class="fu">factor</span>(subtype))</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pheno_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               diagnosis subtype
GSM1116084 Breast cancer   Basal
GSM1116085 Breast cancer   Basal
GSM1116086 Breast cancer    Her2
GSM1116087 Breast cancer   Basal
GSM1116088 Breast cancer    Her2
GSM1116089 Breast cancer    Her2</code></pre>
</div>
</div>
</section>
<section id="feature-data-cleaning" class="level2">
<h2 class="anchored" data-anchor-id="feature-data-cleaning">Feature Data Cleaning</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>missing_geneIDs <span class="ot">&lt;-</span> (feature_data <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="st">`</span><span class="at">ENTREZ_GENE_ID</span><span class="st">`</span> <span class="sc">==</span> <span class="st">""</span>))<span class="sc">$</span>ID</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> feature_data <span class="sc">|&gt;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(<span class="st">"Gene.Symbol"</span>, <span class="st">"Gene.Title"</span>, <span class="st">"ENTREZ_GENE_ID"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">filter</span>(<span class="st">`</span><span class="at">ENTREZ_GENE_ID</span><span class="st">`</span> <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               Gene.Symbol
1007_s_at DDR1 /// MIR4640
1053_at               RFC2
117_at               HSPA6
1294_at   MIR5193 /// UBA7
1316_at               THRA
1405_i_at             CCL5
                                                             Gene.Title
1007_s_at discoidin domain receptor tyrosine kinase 1 /// microRNA 4640
1053_at                     replication factor C (activator 1) 2, 40kDa
117_at                             heat shock 70kDa protein 6 (HSP70B')
1294_at   microRNA 5193 /// ubiquitin-like modifier activating enzyme 7
1316_at                                 thyroid hormone receptor, alpha
1405_i_at                                chemokine (C-C motif) ligand 5
              ENTREZ_GENE_ID
1007_s_at  780 /// 100616237
1053_at                 5982
117_at                  3310
1294_at   7318 /// 100847079
1316_at                 7067
1405_i_at               6352</code></pre>
</div>
</div>
</section>
<section id="gene-expression-data-cleaning" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-data-cleaning">Gene Expression Data Cleaning</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> expr_data <span class="sc">|&gt;</span> <span class="fu">select</span>(<span class="sc">-</span>cell_lines<span class="sc">$</span>geo_accession)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> expr_data[<span class="sc">!</span>(<span class="fu">rownames</span>(expr_data) <span class="sc">%in%</span> missing_geneIDs), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(expr_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          GSM1116084 GSM1116085 GSM1116086 GSM1116087 GSM1116088 GSM1116089
1007_s_at    9.47065    9.67440   10.20800   10.11420   11.16360   10.00690
1053_at      8.36311    8.72194    7.78601    9.44537    7.71242    7.84274
117_at       5.95426    7.02523    6.39671    4.56023    5.29008    6.21590
1294_at      6.02119    7.24581    6.85310    5.42786    7.51120    5.79719
1316_at      3.22997    3.29352    3.26204    3.34766    3.59086    3.40108
1405_i_at   10.82220    9.29455    9.45727   11.51270    8.83075    9.07636
          GSM1116090 GSM1116091 GSM1116092 GSM1116093 GSM1116094 GSM1116095
1007_s_at    9.53932    9.00187    6.54074   10.35510   10.03780   10.64580
1053_at      7.56238    8.01540    9.37874    9.45426    8.09954    8.21743
117_at       6.42180    6.23554    5.43541    3.85652    6.14728    4.53397
1294_at      7.36700    7.17268    7.72472    4.79560    7.16894    6.62358
1316_at      3.27663    3.14212    3.28420    3.27900    3.20978    3.76888
1405_i_at   10.99010   11.15940    9.05848    3.90108   11.01210   10.27710
          GSM1116096 GSM1116097 GSM1116098 GSM1116099 GSM1116100 GSM1116101
1007_s_at   10.44950    8.37233    9.48393   10.14140   10.23970    8.60988
1053_at      8.47638    7.84711    8.50635    7.92972    8.21145    8.33146
117_at       5.34788    5.71406    8.07714    5.91339    4.60687    8.02299
1294_at      6.34457    6.70367    7.73885    5.75826    4.27389    7.41340
1316_at      4.29495    3.25288    3.26944    3.15866    5.43945    3.25189
1405_i_at    9.25481   11.68160   10.81010    9.70204    6.32607   12.07530
          GSM1116102 GSM1116103 GSM1116104 GSM1116105 GSM1116106 GSM1116107
1007_s_at    9.30953   10.11580   10.22820   10.93370   10.45010   10.77400
1053_at      6.61467    8.24303    7.64069    7.84357    6.66872    7.73258
117_at       6.03981    5.21013    6.39396    5.28530    5.93382    6.65636
1294_at      5.75151    7.16413    6.68933    5.86029    7.33326    6.76149
1316_at      3.25374    3.26923    3.31252    5.20997    3.38905    4.68509
1405_i_at    9.49139   11.16260   10.97000    8.03392    9.47221   10.10680
          GSM1116108 GSM1116109 GSM1116110 GSM1116111 GSM1116112 GSM1116113
1007_s_at    9.09858    9.82919    9.76393    9.71049    9.63422    9.32985
1053_at      8.49295    8.04391    9.21119    7.94075    8.60913    9.05156
117_at       7.89833    6.25110    6.20520    6.67730    6.80008    6.02124
1294_at      7.40775    7.80644    7.27217    7.21996    7.01729    5.49913
1316_at      3.38779    3.44371    3.21386    3.46664    3.36358    3.51126
1405_i_at    9.27153   12.08370    9.22943   13.18300    8.84037    8.63069
          GSM1116114 GSM1116115 GSM1116116 GSM1116117 GSM1116118 GSM1116119
1007_s_at    9.13237   10.21230   10.63470   10.19650   10.24290    9.60890
1053_at      7.59912    8.06179    7.61327    7.79566    8.78981    7.23565
117_at       6.51987    5.56796    5.73829    4.46156    6.21806    6.28429
1294_at      7.48972    7.37151    6.31337    6.78600    4.28586    6.41148
1316_at      3.46957    3.50170    3.49541    3.37617    3.29193    3.30101
1405_i_at   10.45860   11.22110    9.28118    8.79246    6.47523    7.82935
          GSM1116120 GSM1116121 GSM1116122 GSM1116123 GSM1116124 GSM1116125
1007_s_at    9.32680   10.51060   10.56970   10.14730    8.34806    9.40561
1053_at      9.35174    7.78091    7.88528    7.39454    8.37482    9.00709
117_at       7.68198    6.02321    5.65715    5.87484    9.14238    6.95742
1294_at      6.12185    7.75981    5.18566    4.53755    7.56331    5.64746
1316_at      3.46059    3.42653    3.27581    3.41536    3.00087    3.50872
1405_i_at   10.30090    9.90023    7.87248    5.74762   11.55320    7.60097
          GSM1116126 GSM1116127 GSM1116128 GSM1116129 GSM1116130 GSM1116131
1007_s_at   10.56410    9.73944   11.17040    9.72613   10.91060    8.60770
1053_at      7.68875    7.48572    8.39529    7.56226    8.89892    9.25822
117_at       3.85010    5.27981    4.83188    6.01448    5.93541    5.89744
1294_at      4.60215    6.02859    5.25828    6.97893    7.25988    6.91073
1316_at      3.35819    3.45760    3.55184    3.53388    3.45018    3.44218
1405_i_at    7.68352    8.54279    8.30117    9.83481   11.94790   11.02720
          GSM1116132 GSM1116133 GSM1116134 GSM1116135 GSM1116136 GSM1116137
1007_s_at   10.38040   10.38110    9.81215    9.46771    9.67632   11.02460
1053_at      8.14164    8.61716    8.19241    9.16920    7.79396    7.37765
117_at       6.18358    5.85304    5.95543    7.70911    6.81969    3.69656
1294_at      7.22719    6.23028    7.06389    6.98866    7.60068    5.94035
1316_at      3.55619    3.43374    3.36207    3.46215    3.43788    3.47543
1405_i_at    9.95795   10.16090    9.73054    9.94269   13.15470    8.46334
          GSM1116138 GSM1116139 GSM1116140 GSM1116141 GSM1116142 GSM1116143
1007_s_at   10.72650    9.75026   10.02410   10.34230   11.15420   11.03680
1053_at      9.03221    8.50789    8.29344    8.46184    8.13731    7.92881
117_at       4.36727    6.00113    7.07612    4.60736    5.55893    5.45299
1294_at      5.39965    6.57596    8.27016    6.90693    6.27358    7.08893
1316_at      3.46998    3.43276    3.40720    3.54718    3.41575    3.53040
1405_i_at    5.83059    9.06461   10.07030    9.54387    9.85074    9.44319
          GSM1116144 GSM1116145 GSM1116146 GSM1116147 GSM1116148 GSM1116149
1007_s_at    9.45143   10.43290   10.34100   10.00610    9.18151    9.82593
1053_at      7.88506    7.60771    8.44041    8.15492    7.85672    8.63647
117_at       5.94825    5.96213    4.84830    5.97218    6.99499    6.10333
1294_at      7.18415    5.49201    8.34065    8.22314    7.66849    4.42144
1316_at      3.47865    3.44486    3.45237    3.66663    3.45032    3.41282
1405_i_at    9.69099   10.88730    6.90163   11.92430   13.12450    4.37651
          GSM1116150 GSM1116151 GSM1116152 GSM1116153 GSM1116154 GSM1116169
1007_s_at   10.60630   10.00750   10.61570   10.40530    9.88982    9.35983
1053_at      8.47677    7.14085    7.82552    7.89417    8.18059    5.64109
117_at       7.45326    6.98528    6.52667    3.65791    5.86942    4.99282
1294_at      7.32962    6.67545    6.29934    3.89667    7.31286    8.90312
1316_at      3.38273    3.51544    3.43433    3.46764    6.05066    5.64136
1405_i_at   10.05190    9.71083    8.17840    4.61677    8.95093    7.72109
          GSM1116170 GSM1116171 GSM1116172 GSM1116173 GSM1116174 GSM1116175
1007_s_at    8.92862    8.51747    9.90086    9.99184    9.80457   10.12430
1053_at      4.97958    5.62614    6.74438    5.70238    6.54442    5.86945
117_at       3.40532    4.93056    4.49811    3.20804    3.37099    3.77733
1294_at      8.82710    7.60378    8.27273    8.02270    8.54070    7.56151
1316_at      4.93700    4.35846    3.49625    4.48971    3.83661    4.31389
1405_i_at    8.30833    6.77847    9.99734    8.51383    7.61147    7.86122
          GSM1116176 GSM1116177 GSM1116178 GSM1116179 GSM1116180 GSM1116181
1007_s_at    9.29848    9.56321    9.96914    9.34603    9.25581    9.65648
1053_at      5.25887    5.95661    5.45792    5.59836    7.42271    7.03281
117_at       4.46238    4.52445    3.14352    4.68036    5.90116    5.01108
1294_at      8.28166    7.74364    8.29586    8.61736    6.39339    7.75967
1316_at      5.43182    4.06242    4.57299    4.32659    3.31515    3.34000
1405_i_at   10.42320    7.95438    8.82569    8.95082    8.82882   10.15770
          GSM1116182 GSM1116183 GSM1116184 GSM1116185 GSM1116186 GSM1116187
1007_s_at   10.69700    9.40894    9.67101   10.23830   10.98810   10.27890
1053_at      6.50217    8.05199    6.70343    7.57718    7.55192    7.39894
117_at       4.74508    4.98127    4.26019    5.84402    8.00253    4.94541
1294_at      8.36900    7.78085    6.79128    7.97305    5.53963    7.72522
1316_at      3.88383    3.77328    3.14845    3.09275    3.09580    3.09989
1405_i_at    9.73859   10.05880    7.23410   10.36570    7.94184    7.28429
          GSM1116188 GSM1116189 GSM1116190 GSM1116191 GSM1116192 GSM1116193
1007_s_at   10.76200   10.89270    9.26293    9.79335   11.46010   10.16940
1053_at      6.64115    7.70086    7.22615    6.79832    7.28488    8.05798
117_at       5.96162    5.87287    5.41029    5.41365    3.95833    7.57243
1294_at      8.20000    5.21315    7.92009    5.79406    6.81766    7.10004
1316_at      3.64292    3.11153    3.42857    3.55201    4.36604    3.47430
1405_i_at   10.00680    7.58043   10.30670    8.30816    5.10893    5.79861
          GSM1116194 GSM1116195 GSM1116196 GSM1116197 GSM1116198 GSM1116199
1007_s_at    9.68111    9.23725   10.37060   10.83540   10.59460   10.99180
1053_at      6.84965    7.29869    7.45002    7.79156    7.68441    6.88320
117_at       5.70363    4.10106    6.13145    5.73012    6.04729    5.63173
1294_at      7.59827    7.23315    7.32834    7.67770    8.44960    8.54789
1316_at      3.25987    3.21713    3.35170    3.37411    4.31510    3.56032
1405_i_at    8.25211    8.69420    8.77493    8.18677    7.95585    7.43344
          GSM1116200 GSM1116201 GSM1116202 GSM1116203 GSM1116204 GSM1116205
1007_s_at   10.79090   10.62090   10.09620   10.75440   10.40590   10.21630
1053_at      6.40945    8.19011    7.51218    7.12197    7.15877    7.06667
117_at       5.61878    4.92947    4.83212    6.28205    5.57738    4.66778
1294_at      7.72629    6.83900    7.69850    8.06425    7.38860    7.66124
1316_at      3.54930    3.86601    3.39656    3.53381    3.42532    3.45824
1405_i_at    8.93008    8.05753   11.08530    9.05865    9.05033    8.42317
          GSM1116206 GSM1116207 GSM1116208 GSM1116209 GSM1116210 GSM1116211
1007_s_at   10.52240   10.86100   10.08800    9.83567   10.87340    9.66704
1053_at      7.51896    8.13146    6.99632    6.67889    7.33159    7.62965
117_at       4.79932    5.16531    4.51390    6.13449    5.10775    7.89241
1294_at      6.96686    6.02907    6.32999    7.39462    7.72787    7.93732
1316_at      2.71935    3.76245    3.40397    3.20680    3.29536    3.28591
1405_i_at    3.10081    8.80312    6.75887    8.75712    9.76065   11.08520
          GSM1116212 GSM1116213 GSM1116214 GSM1116215 GSM1116216 GSM1116217
1007_s_at    9.90649   10.09700   10.62790   10.47760   10.72790   11.10140
1053_at      7.00497    6.99333    8.35771    7.08194    6.84023    7.25026
117_at       6.65585    5.75165    7.25363    6.55543    6.29517    4.81102
1294_at      5.11427    6.85385    6.87828    5.77793    7.78182    7.12204
1316_at      3.22342    3.27224    4.42596    4.64599    3.26699    3.22369
1405_i_at    5.09954    5.54300    8.36752    6.91176    5.68833    7.36957
          GSM1116218 GSM1116219 GSM1116220 GSM1116221 GSM1116222 GSM1116223
1007_s_at   10.33500   10.84180    9.67831   10.70170   10.55280   10.33530
1053_at      7.39641    7.36747    6.91589    5.68589    6.08820    7.35386
117_at       6.32739    6.09316    5.47914    4.42508    6.89790    5.32343
1294_at      7.36391    7.71262    6.74362    7.21228    8.56077    7.24038
1316_at      3.36470    3.52831    3.53458    3.64618    3.34790    3.39311
1405_i_at    7.16191   10.12290    5.93992    6.99717   10.92980    8.09043
          GSM1116224 GSM1116225 GSM1116226 GSM1116227 GSM1116228 GSM1116229
1007_s_at   10.29450   11.09440   10.76850    9.95321   10.22810    9.96811
1053_at      7.20402    7.31024    8.18165    7.38766    6.96242    7.25745
117_at       5.91998    5.48543    5.67256    6.54564    6.46595    8.35832
1294_at      8.17657    7.31011    6.90715    8.54326    7.52390    7.18591
1316_at      3.88391    3.62851    3.55888    3.58091    3.39838    3.50312
1405_i_at    9.85975    4.42757    6.36115   13.06410    9.40301    8.25920
          GSM1116230 GSM1116231 GSM1116232 GSM1116233 GSM1116234 GSM1116235
1007_s_at   10.10680   10.03290    9.84599   10.75850   10.55470    9.45290
1053_at      7.54377    7.41607    5.83454    8.83376    6.88277    6.83700
117_at       6.98743    4.50611    4.78051    4.55847    4.14597    6.10766
1294_at      7.74070    7.44448    7.03001    6.87039    6.37882    6.99674
1316_at      3.57309    3.31572    3.70289    4.82208    3.55309    3.23899
1405_i_at    8.46990    7.32929    7.18265    8.87394    6.36287    9.56499
          GSM1116236 GSM1116237 GSM1116238
1007_s_at   10.80010   10.04640   11.42010
1053_at      7.29371    7.71122    7.63877
117_at       5.41169    4.57742    4.67159
1294_at      8.37598    6.56774    7.40374
1316_at      3.84879    3.28091    3.24882
1405_i_at    8.11443    8.07087    8.44304</code></pre>
</div>
</div>
</section>
<section id="save-clean-data" class="level2">
<h2 class="anchored" data-anchor-id="save-clean-data">Save Clean Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(expr_data, <span class="st">"../data/GSE45827_clean_expression.csv"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(pheno_data, <span class="st">"../data/GSE45827_clean_Phenotype_metadata.csv"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(feature_data, <span class="st">"../data/GSE45827_clean_Feature_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="describe-data" class="level1">
<h1>Describe Data</h1>
<section id="load-libraries-2" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-2">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data-2" class="level2">
<h2 class="anchored" data-anchor-id="load-data-2">Load Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_clean_expression.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_clean_Feature_metadata.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_clean_Phenotype_metadata.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-expression-data" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-data">Gene Expression Data</h2>
<p>Columns: Samples</p>
<p>Rows: Genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>expr_data[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             GSM1116084 GSM1116085 GSM1116086 GSM1116087 GSM1116088 GSM1116089
1007_s_at       9.47065    9.67440   10.20800   10.11420   11.16360   10.00690
1053_at         8.36311    8.72194    7.78601    9.44537    7.71242    7.84274
117_at          5.95426    7.02523    6.39671    4.56023    5.29008    6.21590
1294_at         6.02119    7.24581    6.85310    5.42786    7.51120    5.79719
1316_at         3.22997    3.29352    3.26204    3.34766    3.59086    3.40108
1405_i_at      10.82220    9.29455    9.45727   11.51270    8.83075    9.07636
1438_at         4.39698    6.68936    5.46440    5.29748    6.81821    5.10469
1487_at         6.65320    7.20820    6.68974    6.87276    6.45744    6.87822
1552256_a_at   10.91320    9.32204    7.73131   10.39210    7.05836    7.89627
1552257_a_at   10.07510    9.34476    8.43573    9.63893    9.37463    8.27604
             GSM1116090 GSM1116091 GSM1116092 GSM1116093
1007_s_at       9.53932    9.00187    6.54074   10.35510
1053_at         7.56238    8.01540    9.37874    9.45426
117_at          6.42180    6.23554    5.43541    3.85652
1294_at         7.36700    7.17268    7.72472    4.79560
1316_at         3.27663    3.14212    3.28420    3.27900
1405_i_at      10.99010   11.15940    9.05848    3.90108
1438_at         5.05443    6.87350    5.20408    8.83710
1487_at         6.67559    6.31095    6.88597    6.93847
1552256_a_at    8.51650    9.34351    8.62752   10.22550
1552257_a_at    7.45211    9.68665    8.50533    9.30071</code></pre>
</div>
</div>
<p><strong>Number of Genes</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(expr_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 26468</code></pre>
</div>
</div>
<p><strong>Number of Samples</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(expr_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 141</code></pre>
</div>
</div>
</section>
<section id="feature-data-description" class="level2">
<h2 class="anchored" data-anchor-id="feature-data-description">Feature Data Description</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               Gene.Symbol
1007_s_at DDR1 /// MIR4640
1053_at               RFC2
117_at               HSPA6
1294_at   MIR5193 /// UBA7
1316_at               THRA
1405_i_at             CCL5
                                                             Gene.Title
1007_s_at discoidin domain receptor tyrosine kinase 1 /// microRNA 4640
1053_at                     replication factor C (activator 1) 2, 40kDa
117_at                             heat shock 70kDa protein 6 (HSP70B')
1294_at   microRNA 5193 /// ubiquitin-like modifier activating enzyme 7
1316_at                                 thyroid hormone receptor, alpha
1405_i_at                                chemokine (C-C motif) ligand 5
              ENTREZ_GENE_ID
1007_s_at  780 /// 100616237
1053_at                 5982
117_at                  3310
1294_at   7318 /// 100847079
1316_at                 7067
1405_i_at               6352</code></pre>
</div>
</div>
<p><strong>5 Columns:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(feature_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Gene.Symbol"    "Gene.Title"     "ENTREZ_GENE_ID"</code></pre>
</div>
</div>
<ul>
<li><p><strong>Representative.Public.ID</strong>: Probe ID from the microarray platform</p></li>
<li><p><strong>Gene.Symbol</strong>: The official gene symbol, following HUGO Gene Nomenclature Committee (HGNC) naming</p></li>
<li><p><strong>Gene.Title</strong>: A longer, human-readable description of the gene.</p></li>
<li><p><strong>ENTREZ_GENE_ID</strong>: The NCBI Entrez Gene identifier (A unique numeric ID for each gene)</p></li>
<li><p><strong>RefSeq.Transcript.ID</strong>: One or more RefSeq transcript accession numbers that correspond to the gene or probe.</p></li>
</ul>
<p><strong>Number of Genes</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(feature_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 26468</code></pre>
</div>
</div>
</section>
<section id="phenotype-data-description" class="level2">
<h2 class="anchored" data-anchor-id="phenotype-data-description">Phenotype Data Description</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pheno_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               diagnosis subtype
GSM1116084 Breast cancer   Basal
GSM1116085 Breast cancer   Basal
GSM1116086 Breast cancer    Her2
GSM1116087 Breast cancer   Basal
GSM1116088 Breast cancer    Her2
GSM1116089 Breast cancer    Her2</code></pre>
</div>
</div>
<p><strong>2 Columns:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pheno_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "diagnosis" "subtype"  </code></pre>
</div>
</div>
<ul>
<li><p><strong>Diagnosis</strong>: Breast cancer or healthy sample</p></li>
<li><p><strong>Subtype</strong>: Breast cancer subtype</p></li>
</ul>
<p><strong>Number of Samples</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(pheno_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 141</code></pre>
</div>
</div>
<p><strong>Distribution of Diagnosis Values</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="sc">|&gt;</span> <span class="fu">count</span>(diagnosis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      diagnosis   n
1 Breast cancer 130
2       Healthy  11</code></pre>
</div>
</div>
<p><strong>Distribution of Cancer Subtype Values</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="sc">|&gt;</span> <span class="fu">count</span>(subtype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    subtype  n
1     Basal 41
2   Healthy 11
3      Her2 30
4 Luminal A 29
5 Luminal B 30</code></pre>
</div>
</div>
</section>
</section>
<section id="pca-analysis" class="level1">
<h1>PCA Analysis</h1>
<section id="load-libraries-3" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-3">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbiplot)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data-3" class="level2">
<h2 class="anchored" data-anchor-id="load-data-3">Load Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_clean_expression.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/GSE45827_clean_Phenotype_metadata.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pca-calculation-and-plotting" class="level2">
<h2 class="anchored" data-anchor-id="pca-calculation-and-plotting">PCA Calculation And Plotting</h2>
<p><strong>Format data for PCA</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>transp_expr <span class="ot">&lt;-</span> <span class="fu">t</span>(expr_data) <span class="sc">|&gt;</span> <span class="fu">data.frame</span>()</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>transp_expr <span class="ot">&lt;-</span> transp_expr <span class="sc">|&gt;</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diagnosis =</span> pheno_data<span class="sc">$</span>diagnosis)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>transp_expr <span class="ot">&lt;-</span> transp_expr <span class="sc">|&gt;</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">subtype =</span> pheno_data<span class="sc">$</span>subtype)</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>expr_only <span class="ot">&lt;-</span> transp_expr <span class="sc">|&gt;</span>  <span class="fu">select</span>(<span class="sc">-</span>subtype, <span class="sc">-</span>diagnosis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Perform PCA</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>pc <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(expr_only, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>PCA: Cancer vs Healthy</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbiplot</span>(pc, </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">obs.scale =</span> <span class="dv">1</span>,</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">var.scale =</span> <span class="dv">1</span>,</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">groups =</span> transp_expr<span class="sc">$</span>diagnosis,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">var.axes =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-31-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>PCA: By Cancer Subtype</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbiplot</span>(pc, </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">obs.scale =</span> <span class="dv">1</span>,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">var.scale =</span> <span class="dv">1</span>,</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">groups =</span> transp_expr<span class="sc">$</span>subtype,</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">var.axes =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-32-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="data-analysis" class="level2">
<h2 class="anchored" data-anchor-id="data-analysis">Data Analysis</h2>
<p>PCA Plots are a tool for visualizing high dimensional data onto an orthogonal basis so that the first principal component (PC, aka the axis) is aligned with the largest source of variance, and then the second PC is aligned with the second largest source of variance etc. This is used to visualize and identify underlying patterns within the data.</p>
<p>The second plot shows us which samples are similar in terms of their gene expression, and which ones are different, which reveals clusters of similarly behaving gene expression. As the coloring shows, the tumour subtypes are plotted in certain area of the graph. This means, there are patterns of gene expression within one subtype that differ generally from other subtypes. This proves clinic significance of the subtypes and enforces further analysis of the gene expression behavior.</p>
<p>The first graph shows the more general plot and instead of plotting for all the different subtypes of tumor, we are looking at the genes that are highly expressed in all the breast cancers vs in normal cells, and colored based on these two variables.</p>
</section>
</section>
<section id="differential-gene-expression-analysis" class="level1">
<h1>Differential Gene Expression Analysis</h1>
<section id="load-libraries-4" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-4">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data-4" class="level2">
<h2 class="anchored" data-anchor-id="load-data-4">Load Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_expression.csv"</span>,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="dv">1</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_Feature_metadata.csv"</span>,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="dv">1</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_Phenotype_metadata.csv"</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="dv">1</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="linear-modeling-and-differential-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="linear-modeling-and-differential-expression-analysis">Linear Modeling and Differential Expression Analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pheno_data<span class="sc">$</span>subtype)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> groups)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>clean_levels <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">levels</span>(groups))</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> clean_levels</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(expr_data, design)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>contrast_matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Basal_vs_Her2 =</span> Basal <span class="sc">-</span> Her2,</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Basal_vs_LumA =</span> Basal <span class="sc">-</span> Luminal.A,</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">Her2_vs_LumA =</span> Her2 <span class="sc">-</span> Luminal.A,</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">LumA_vs_LumB =</span> Luminal.A <span class="sc">-</span> Luminal.B,</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels =</span> design</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>fit_contrast <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contrast_matrix)</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>fit_ebayes <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit_contrast)</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(<span class="fu">decideTests</span>(fit_ebayes)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       Basal_vs_Her2 Basal_vs_LumA Her2_vs_LumA LumA_vs_LumB
Down            4214          7551         5390         3478
NotSig         17359         10767        15936        19817
Up              4895          8150         5142         3173</code></pre>
</div>
</div>
</section>
<section id="preparing-the-gene-list" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-gene-list">Preparing the Gene List</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>top_table <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_ebayes, <span class="at">coef =</span> <span class="st">"Basal_vs_LumA"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               logFC  AveExpr        t      P.Value    adj.P.Val         B
210052_s_at 5.406121 7.369013 22.76376 9.494968e-49 2.513128e-44 100.32266
209408_at   5.077893 6.022339 21.79878 1.059288e-46 1.401861e-42  95.67803
209464_at   4.139424 5.469734 21.15724 2.595258e-45 2.289710e-41  92.52381
206364_at   4.744102 7.470250 21.06249 4.180713e-45 2.766378e-41  92.05345
219918_s_at 5.210790 7.748509 20.18735 3.606759e-43 1.909274e-39  87.65373
204962_s_at 4.973160 6.729938 20.14428 4.502934e-43 1.986394e-39  87.43458</code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>top_table<span class="sc">$</span>probe_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_table)</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>probe_to_gene_map <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">probe_id =</span> <span class="fu">rownames</span>(feature_data),</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_symbol =</span> feature_data<span class="sc">$</span><span class="st">"Gene.Symbol"</span> )</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>res_with_symbols <span class="ot">&lt;-</span> <span class="fu">merge</span>(top_table, probe_to_gene_map, <span class="at">by=</span><span class="st">"probe_id"</span>)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>res_with_symbols <span class="ot">&lt;-</span> res_with_symbols <span class="sc">%&gt;%</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(t) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(gene_symbol) <span class="sc">&amp;</span> gene_symbol <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>res_ranked <span class="ot">&lt;-</span> res_with_symbols <span class="sc">%&gt;%</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> <span class="fu">abs</span>(t), <span class="at">n =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> res_ranked<span class="sc">$</span>t</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(stats) <span class="ot">&lt;-</span> res_ranked<span class="sc">$</span>gene_symbol</span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">sort</span>(stats, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(stats)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             TPX2             KIF2C             AURKB             KIF14 
         22.76376          21.79878          21.15724          21.06249 
             ASPM CENPA /// SLC35F6 
         20.18735          20.14428 </code></pre>
</div>
</div>
</section>
<section id="running-gsea" class="level2">
<h2 class="anchored" data-anchor-id="running-gsea">Running GSEA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>BP_df <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"human"</span>, <span class="at">category =</span> <span class="st">"C5"</span>, <span class="at">subcategory =</span> <span class="st">"BP"</span>)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">split</span>(BP_df<span class="sc">$</span>gene_symbol, <span class="at">f =</span> BP_df<span class="sc">$</span>gs_name)</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgseaMultilevel</span>(</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">pathways =</span> pathways,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">stats    =</span> stats,</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">minSize  =</span> <span class="dv">15</span>,</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxSize  =</span> <span class="dv">500</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Top GSEA results (Basal_vs_LumA):"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Top GSEA results (Basal_vs_LumA):"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fgseaRes[<span class="fu">order</span>(padj), ])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                     pathway         pval         padj  log2err
                                      &lt;char&gt;        &lt;num&gt;        &lt;num&gt;    &lt;num&gt;
1:               GOBP_CHROMOSOME_SEGREGATION 8.820515e-40 3.181560e-36 1.640742
2:                    GOBP_ORGANELLE_FISSION 7.222175e-38 1.302519e-34 1.602431
3:       GOBP_NUCLEAR_CHROMOSOME_SEGREGATION 3.936871e-36 4.733431e-33 1.563182
4:             GOBP_MITOTIC_NUCLEAR_DIVISION 1.895577e-32 1.709337e-29 1.481568
5:         GOBP_SISTER_CHROMATID_SEGREGATION 1.473758e-31 1.063169e-28 1.460452
6: GOBP_MITOTIC_SISTER_CHROMATID_SEGREGATION 4.080198e-30 2.452879e-27 1.424563
          ES      NES  size  leadingEdge
       &lt;num&gt;    &lt;num&gt; &lt;int&gt;       &lt;list&gt;
1: 0.6064707 3.238701   273 KIF2C, A....
2: 0.5389512 3.006860   382 TPX2, KI....
3: 0.6262546 3.232082   219 KIF2C, A....
4: 0.5812431 3.066768   254 TPX2, KI....
5: 0.6462065 3.218662   173 KIF2C, A....
6: 0.6635166 3.224277   150 KIF2C, A....</code></pre>
</div>
</div>
</section>
<section id="the-enrichment-plot" class="level2">
<h2 class="anchored" data-anchor-id="the-enrichment-plot">The Enrichment Plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>fgseaRes_ordered <span class="ot">&lt;-</span> fgseaRes[<span class="fu">order</span>(padj), ]</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>top_pathway_name <span class="ot">&lt;-</span> fgseaRes_ordered<span class="sc">$</span>pathway[<span class="dv">1</span>]</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>leading_edge_genes <span class="ot">&lt;-</span> fgseaRes_ordered<span class="sc">$</span>leadingEdge[[<span class="dv">1</span>]]</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>first_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(leading_edge_genes,<span class="dv">10</span>)</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(first_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "KIF2C"  "AURKB"  "KIF14"  "CEP55"  "BUB1"   "FAM83D" "CDC20"  "TTK"   
 [9] "MKI67"  "CENPF" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotEnrichment</span>(pathways[[top_pathway_name]],</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">stats =</span> stats) <span class="sc">+</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> top_pathway_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="00_all_files/figure-html/unnamed-chunk-38-1.svg" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="getting-the-top-genes" class="level2">
<h2 class="anchored" data-anchor-id="getting-the-top-genes">Getting the Top Genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>top_5_gene_lists <span class="ot">&lt;-</span> fgseaRes_ordered<span class="sc">$</span>leadingEdge[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>top_5_names <span class="ot">&lt;-</span> fgseaRes_ordered<span class="sc">$</span>pathway[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>) {</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">head</span>(top_5_gene_lists[[i]], <span class="dv">10</span>))</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "KIF2C"  "AURKB"  "KIF14"  "CEP55"  "BUB1"   "FAM83D" "CDC20"  "TTK"   
 [9] "MKI67"  "CENPF" 
 [1] "TPX2"  "KIF2C" "AURKB" "KIF14" "ASPM"  "CEP55" "BUB1"  "CDC20" "TTK"  
[10] "MKI67"
 [1] "KIF2C"  "AURKB"  "KIF14"  "CEP55"  "BUB1"   "FAM83D" "CDC20"  "TTK"   
 [9] "CENPF"  "CDCA5" 
 [1] "TPX2"  "KIF2C" "AURKB" "KIF14" "CEP55" "BUB1"  "CDC20" "TTK"   "MKI67"
[10] "CENPF"
 [1] "KIF2C" "AURKB" "KIF14" "CEP55" "BUB1"  "CDC20" "TTK"   "CENPF" "CDCA5"
[10] "NDC80"</code></pre>
</div>
</div>
</section>
<section id="intersection-analysis-of-core-enrichment-genes" class="level2">
<h2 class="anchored" data-anchor-id="intersection-analysis-of-core-enrichment-genes">Intersection Analysis of Core Enrichment Genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>core_drivers <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, top_5_gene_lists)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>all_genes_flat <span class="ot">&lt;-</span> <span class="fu">unlist</span>(top_5_gene_lists)      </span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(all_genes_flat)      </span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>gene_counts_sorted <span class="ot">&lt;-</span> <span class="fu">sort</span>(gene_counts, <span class="at">decreasing =</span> <span class="cn">TRUE</span>) </span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> gene_counts_sorted[gene_counts_sorted <span class="sc">&gt;=</span> <span class="dv">3</span>]</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>res</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>all_genes_flat
  AURKB   BAZ1B    BUB1   BUB1B   CCNB1   CDC20    CDC6   CDCA5   CDCA8    CDT1 
      5       5       5       5       5       5       5       5       5       5 
  CENPE   CENPF   CENPK   CEP55  CHMP4B   CHTF8  DLGAP5   DSCC1    DSN1    EML4 
      5       5       5       5       5       5       5       5       5       5 
  ESPL1   FBXO5    GEN1  KATNB1   KIF14  KIF18A  KIF18B   KIF23   KIF2C   KIF4A 
      5       5       5       5       5       5       5       5       5       5 
  KIFC1  KNSTRN   KNTC1  MAD2L1  MAD2L2   NAA10   NAA50  NCAPD2  NCAPD3   NCAPG 
      5       5       5       5       5       5       5       5       5       5 
 NCAPG2   NCAPH  NCAPH2   NDC80    NEK2    NUF2   NUP62  NUSAP1    PLK1    PRC1 
      5       5       5       5       5       5       5       5       5       5 
  PSRC1   PTTG1 RACGAP1   RAD21     RAN    RRS1   SEH1L   SMC1A    SMC2    SMC4 
      5       5       5       5       5       5       5       5       5       5 
  SPAG5   SPDL1   TACC3  TRIP13     TTK   TUBG1   UBE2C   VPS4A  ZWILCH   ZWINT 
      5       5       5       5       5       5       5       5       5       5 
HORMAD1   TOP2A   ACTR2   ACTR3   BRIP1   CCNE1   CCNE2   CDCA2   DDX11    EME1 
      4       4       3       3       3       3       3       3       3       3 
 FANCD2    FEN1   MCMBP   MKI67    NDC1    NDE1 PSMC3IP    RCC1    RMI1    RMI2 
      3       3       3       3       3       3       3       3       3       3 </code></pre>
</div>
</div>
</section>
<section id="saving-results" class="level2">
<h2 class="anchored" data-anchor-id="saving-results">Saving Results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>fgseaRes_flat <span class="ot">&lt;-</span> fgseaRes</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>fgseaRes_flat<span class="sc">$</span>leadingEdge <span class="ot">&lt;-</span> <span class="fu">sapply</span>(fgseaRes_flat<span class="sc">$</span>leadingEdge, paste, <span class="at">collapse =</span> <span class="st">"; "</span>)</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(fgseaRes_flat, <span class="st">"../data/Top_GSEA_results_Basal_vs_LumA.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>top_5_gene_df <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(top_5_gene_lists, <span class="sc">~</span><span class="fu">as.data.frame</span>(<span class="fu">t</span>(.x), <span class="at">.id=</span><span class="st">"id"</span>))</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(top_5_gene_df, <span class="st">"../data/Genes_from_Top_5_pathways.csv"</span>)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res, <span class="st">"../data/Intersected_Genes.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) </span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>my_plot <span class="ot">&lt;-</span> <span class="fu">plotEnrichment</span>(pathways[[top_pathway_name]], <span class="at">stats =</span> stats) <span class="sc">+</span></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>           <span class="fu">labs</span>(<span class="at">title =</span> top_pathway_name)</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"../results/Intersected_Genes_Plot.png"</span>, </span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> my_plot, </span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">8</span>, </span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="network-analysis" class="level1">
<h1>Network Analysis</h1>
<section id="load-libraries-5" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-5">Load Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GEOquery"</span>)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"Biobase"</span>)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"limma"</span>)</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tibble"</span>)</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"stringr"</span>)</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"readr"</span>)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"igraph"</span>)</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msigdbr"</span>)</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"clusterProfiler"</span>)</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"Matrix"</span>)</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"quarto"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data-5" class="level2">
<h2 class="anchored" data-anchor-id="load-data-5">Load Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>expr_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_expression.csv"</span>,</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">row.names =</span> <span class="dv">1</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>feature_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_Feature_metadata.csv"</span></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(feature_data)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"ProbeID"</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../data/GSE45827_clean_Phenotype_metadata.csv"</span>,</span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pheno_data)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"SampleID"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Set output directory</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">"../results"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-expression-analysis" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis">Differential Expression Analysis</h2>
<p>Analysis is performed using limma. It involves a linear model that is fitted for the Tumor vs Normal contrast and empirical Bayes shrinkage which is applied to stabilize variance estimates. All tested genes are carried forward to the network analysis, while significantly differentiated ones (e.g.&nbsp;FDR &lt; 0.05 and |logFC| &gt; 1) are later used as seed genes to anchor subnetworks.</p>
<p><strong>Set up Model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>design_all <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> diagnosis,</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> pheno_data</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Fit model</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>fit_all <span class="ot">&lt;-</span> expr_data <span class="sc">|&gt;</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lmFit</span>(design_all) <span class="sc">|&gt;</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">eBayes</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Get Results</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>tt_tumor_vs_normal <span class="ot">&lt;-</span> <span class="fu">topTable</span>(</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  fit_all,</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">coef   =</span> <span class="st">"diagnosisHealthy"</span>,</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">number =</span> <span class="cn">Inf</span>,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">sort.by =</span> <span class="st">"P"</span></span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tt_tumor_vs_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               logFC  AveExpr        t      P.Value    adj.P.Val        B
211565_at   6.357343 2.569430 52.42703 9.801966e-95 2.594384e-90 202.6620
243689_s_at 6.101876 3.115981 37.98376 3.261078e-76 4.315711e-72 162.1877
224012_at   4.961053 3.268422 31.44882 8.169992e-66 7.208112e-62 138.9671
238062_at   4.518850 2.634483 27.33766 2.118232e-58 1.401634e-54 122.2649
213715_s_at 2.845828 5.467643 27.16630 4.483468e-58 2.373368e-54 121.5290
1562527_at  5.331220 2.888688 26.74320 2.895354e-57 1.277237e-53 119.6975</code></pre>
</div>
</div>
<p><strong>View Differential Expression Results</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>tt_tumor_vs_normal <span class="ot">&lt;-</span> tt_tumor_vs_normal <span class="sc">|&gt;</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ProbeID"</span>) <span class="sc">|&gt;</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>    feature_data <span class="sc">|&gt;</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(</span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>        ProbeID,</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">Gene.Symbol =</span> <span class="st">`</span><span class="at">Gene.Symbol</span><span class="st">`</span></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"ProbeID"</span></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Gene.Symbol, <span class="at">.after =</span> ProbeID)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tt_tumor_vs_normal)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      ProbeID                                         Gene.Symbol    logFC
1   211565_at                                              SH3GL3 6.357343
2 243689_s_at                                               FRG1B 6.101876
3   224012_at ANKRD20A1 /// ANKRD20A2 /// ANKRD20A3 /// ANKRD20A4 4.961053
4   238062_at                                             GPIHBP1 4.518850
5 213715_s_at                                               KANK3 2.845828
6  1562527_at                                           LOC441666 5.331220
   AveExpr        t      P.Value    adj.P.Val        B
1 2.569430 52.42703 9.801966e-95 2.594384e-90 202.6620
2 3.115981 37.98376 3.261078e-76 4.315711e-72 162.1877
3 3.268422 31.44882 8.169992e-66 7.208112e-62 138.9671
4 2.634483 27.33766 2.118232e-58 1.401634e-54 122.2649
5 5.467643 27.16630 4.483468e-58 2.373368e-54 121.5290
6 2.888688 26.74320 2.895354e-57 1.277237e-53 119.6975</code></pre>
</div>
</div>
<p><strong>Number of significant genes (FDR&lt;0.05, |logFC|&gt;1):</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>    tt_tumor_vs_normal<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">abs</span>(tt_tumor_vs_normal<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 8134</code></pre>
</div>
</div>
<p><strong>Save Results</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>DE_results <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tumor_vs_Normal =</span> tt_tumor_vs_normal</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="network-analysis-1" class="level2">
<h2 class="anchored" data-anchor-id="network-analysis-1">Network Analysis</h2>
<p><strong>Load Helper Functions</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"99_proj_func.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="network-construction-and-summary" class="level2">
<h2 class="anchored" data-anchor-id="network-construction-and-summary">Network Construction and Summary</h2>
<p>We construct a Reactome co-membership network using genepathway incidence data. The edges reflect shared pathway membership and seed genes serve as anchors for selecting relevant subnetworks. After Louvain community detection, we extract the largest connected clusters and compute core network statistics including modules, hubs, and connectivity.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>tag <span class="ot">&lt;-</span> <span class="st">"Tumor_vs_Normal"</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>tt_gene <span class="ot">&lt;-</span> <span class="fu">collapse_to_genes</span>(</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>  DE_results[[tag]],</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  feature_data</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>universe_symbols <span class="ot">&lt;-</span> tt_gene<span class="sc">$</span>symbol</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(universe_symbols)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "A1BG"     "A1BG-AS1" "A2M"      "A2M-AS1"  "A2ML1"    "A4GALT"  </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>sym2 <span class="ot">&lt;-</span> tt_gene <span class="sc">|&gt;</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>    symbol,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>    logFC,</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>    adj.P.Val</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sym2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  3
  symbol     logFC adj.P.Val
  &lt;chr&gt;      &lt;dbl&gt;     &lt;dbl&gt;
1 A1BG      0.103  0.492    
2 A1BG-AS1  0.174  0.160    
3 A2M       1.41   0.0000117
4 A2M-AS1   0.134  0.557    
5 A2ML1    -0.762  0.222    
6 A4GALT    0.0642 0.377    </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>seeds <span class="ot">&lt;-</span> <span class="fu">pick_seeds</span>(</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  tt_gene,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fdr =</span> <span class="fl">0.01</span>,</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">lfc =</span> <span class="fl">1.5</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(seeds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SH3GL3"                                             
[2] "FRG1B"                                              
[3] "ANKRD20A1 /// ANKRD20A2 /// ANKRD20A3 /// ANKRD20A4"
[4] "GPIHBP1"                                            
[5] "KANK3"                                              
[6] "LOC441666"                                          </code></pre>
</div>
</div>
<p><strong>Universe genes:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(universe_symbols)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 14712</code></pre>
</div>
</div>
<p><strong>Seed genes:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(seeds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3626</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>reactome_df <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="st">"Homo sapiens"</span>,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">category =</span> <span class="st">"C2"</span>,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">subcategory =</span> <span class="st">"REACTOME"</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>    gs_name,</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>    gene_symbol</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">|&gt;</span></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>    gene_symbol <span class="sc">%in%</span> universe_symbols</span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(reactome_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  2
  gs_name                         gene_symbol
  &lt;chr&gt;                           &lt;chr&gt;      
1 REACTOME_2_LTR_CIRCLE_FORMATION BANF1      
2 REACTOME_2_LTR_CIRCLE_FORMATION HMGA1      
3 REACTOME_2_LTR_CIRCLE_FORMATION LIG4       
4 REACTOME_2_LTR_CIRCLE_FORMATION PSIP1      
5 REACTOME_2_LTR_CIRCLE_FORMATION XRCC4      
6 REACTOME_2_LTR_CIRCLE_FORMATION XRCC5      </code></pre>
</div>
</div>
<p><strong>Reactome pathways covering universe:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(reactome_df<span class="sc">$</span>gs_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1608</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(reactome_df<span class="sc">$</span>gene_symbol))</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>paths <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(reactome_df<span class="sc">$</span>gs_name))</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>gi <span class="ot">&lt;-</span> <span class="fu">match</span>(reactome_df<span class="sc">$</span>gene_symbol, genes)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>pi <span class="ot">&lt;-</span> <span class="fu">match</span>(reactome_df<span class="sc">$</span>gs_name, paths)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Make Adjacency Matrix</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>Inc <span class="ot">&lt;-</span> <span class="fu">sparseMatrix</span>(</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> gi,</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">j =</span> pi,</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="dv">1</span>,</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dims =</span> <span class="fu">c</span>(<span class="fu">length</span>(genes), <span class="fu">length</span>(paths)),</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">dimnames =</span> <span class="fu">list</span>(genes, paths)</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a>Adj <span class="ot">&lt;-</span> Inc <span class="sc">%*%</span> <span class="fu">t</span>(Inc)</span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a><span class="fu">diag</span>(Adj) <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>min_shared <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>seed_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">rownames</span>(Adj) <span class="sc">%in%</span> seeds)</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>nbr_raw <span class="ot">&lt;-</span> Adj[seed_idx, , drop <span class="ot">=</span> <span class="cn">FALSE</span>] <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(seed_idx) <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>  nbr_idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">as.vector</span>(nbr_raw))</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>  nbr_idx <span class="ot">&lt;-</span> <span class="fu">unique</span>(</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlist</span>(</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">apply</span>(</span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>        nbr_raw,</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a>        <span class="dv">1</span>,</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a>        which</span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>nbr_idx <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(seed_idx, nbr_idx))</span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a>Adj_sub <span class="ot">&lt;-</span> Adj[nbr_idx, nbr_idx, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">summary</span>(Adj_sub)</span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> A[</span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a>  A<span class="sc">$</span>i <span class="sc">&lt;</span> A<span class="sc">$</span>j <span class="sc">&amp;</span> A<span class="sc">$</span>x <span class="sc">&gt;=</span> min_shared,</span>
<span id="cb95-39"><a href="#cb95-39" aria-hidden="true" tabindex="-1"></a>  ,</span>
<span id="cb95-40"><a href="#cb95-40" aria-hidden="true" tabindex="-1"></a>  drop <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb95-41"><a href="#cb95-41" aria-hidden="true" tabindex="-1"></a>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Define Edges</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">from =</span> <span class="fu">rownames</span>(Adj_sub)[A<span class="sc">$</span>i],</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">to =</span> <span class="fu">rownames</span>(Adj_sub)[A<span class="sc">$</span>j],</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight =</span> <span class="fu">as.numeric</span>(A<span class="sc">$</span>x)</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Define Vertices</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>node_syms <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">c</span>(edges<span class="sc">$</span>from, edges<span class="sc">$</span>to)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Define Graph</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">d =</span> edges,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">vertices =</span> <span class="fu">tibble</span>(<span class="at">symbol =</span> node_syms),</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">directed =</span> <span class="cn">FALSE</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">V</span>(g)<span class="sc">$</span>name</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>seed <span class="ot">&lt;-</span> <span class="fu">V</span>(g)<span class="sc">$</span>symbol <span class="sc">%in%</span> seeds</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>logFC <span class="ot">&lt;-</span> sym2<span class="sc">$</span>logFC[</span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">match</span>(<span class="fu">V</span>(g)<span class="sc">$</span>symbol, sym2<span class="sc">$</span>symbol)</span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>adj.P.Val <span class="ot">&lt;-</span> sym2<span class="sc">$</span>adj.P.Val[</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">match</span>(<span class="fu">V</span>(g)<span class="sc">$</span>symbol, sym2<span class="sc">$</span>symbol)</span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">simplify</span>(</span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a>  g,</span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">remove.multiple =</span> <span class="cn">TRUE</span>,</span>
<span id="cb98-21"><a href="#cb98-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">remove.loops =</span> <span class="cn">TRUE</span>,</span>
<span id="cb98-22"><a href="#cb98-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">edge.attr.comb =</span> <span class="fu">list</span>(<span class="at">weight =</span> <span class="st">"sum"</span>)</span>
<span id="cb98-23"><a href="#cb98-23" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Get Clusters</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>com <span class="ot">&lt;-</span> <span class="fu">cluster_louvain</span>(</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  g,</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">weights =</span> <span class="fu">E</span>(g)<span class="sc">$</span>weight</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(g)<span class="sc">$</span>module <span class="ot">&lt;-</span> <span class="fu">membership</span>(com)</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">components</span>(g)</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>gc_id <span class="ot">&lt;-</span> <span class="fu">which.max</span>(comp<span class="sc">$</span>csize)</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>  g,</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">vids =</span> <span class="fu">which</span>(comp<span class="sc">$</span>membership <span class="sc">==</span> gc_id)</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Final network - Nodes:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gorder</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7231</code></pre>
</div>
</div>
<p><strong>Edges:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gsize</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1379925</code></pre>
</div>
</div>
<p><strong>Modules detected:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>module))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_net_modules_hubs</span>(</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  g,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  tag,</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>  output_dir</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>svg 
  2 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>hub_logical <span class="ot">&lt;-</span> <span class="fu">get_hub_logical</span>(g, <span class="at">q =</span> <span class="fl">0.90</span>)</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hub_logical)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> A1BG   A2M  AAAS  AACS AADAT  AAK1 
FALSE FALSE  TRUE FALSE FALSE FALSE </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>nodes_out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">symbol =</span> <span class="fu">V</span>(g)<span class="sc">$</span>symbol,</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">hub =</span> hub_logical,</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">seed =</span> <span class="fu">V</span>(g)<span class="sc">$</span>seed,</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">module =</span> <span class="fu">V</span>(g)<span class="sc">$</span>module,</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">degree =</span> <span class="fu">degree</span>(g),</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">strength =</span> <span class="fu">strength</span>(</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>    g,</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">vids =</span> <span class="fu">V</span>(g),</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">E</span>(g)<span class="sc">$</span>weight</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb110-12"><a href="#cb110-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">betweenness =</span> <span class="fu">betweenness</span>(</span>
<span id="cb110-13"><a href="#cb110-13" aria-hidden="true" tabindex="-1"></a>    g,</span>
<span id="cb110-14"><a href="#cb110-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">directed =</span> <span class="cn">FALSE</span>,</span>
<span id="cb110-15"><a href="#cb110-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">normalized =</span> <span class="cn">TRUE</span></span>
<span id="cb110-16"><a href="#cb110-16" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb110-17"><a href="#cb110-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">logFC =</span> <span class="fu">V</span>(g)<span class="sc">$</span>logFC,</span>
<span id="cb110-18"><a href="#cb110-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">adj.P.Val =</span> <span class="fu">V</span>(g)<span class="sc">$</span>adj.P.Val</span>
<span id="cb110-19"><a href="#cb110-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb110-20"><a href="#cb110-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-21"><a href="#cb110-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nodes_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  9
  symbol hub   seed  module degree strength betweenness  logFC adj.P.Val
  &lt;chr&gt;  &lt;lgl&gt; &lt;lgl&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;       &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;
1 A1BG   FALSE FALSE      1    588     1392  0.0000380   0.103 0.492    
2 A2M    FALSE FALSE      2    409     1006  0.000360    1.41  0.0000117
3 AAAS   TRUE  FALSE      3   1902     7631  0.00561    -0.177 0.332    
4 AACS   FALSE FALSE      4      7       20  0           0.496 0.0719   
5 AADAT  FALSE FALSE      4     16       32  0.0000330  -0.366 0.245    
6 AAK1   FALSE FALSE      5    553     1324  0.00000646  0.560 0.000544 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>mod_DE_info <span class="ot">&lt;-</span> nodes_out <span class="sc">|&gt;</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(module) <span class="sc">|&gt;</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">module_size =</span> <span class="fu">n</span>(),</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">n_DE =</span> <span class="fu">sum</span>(</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>      adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>,</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mod_DE_info)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  3
  module module_size  n_DE
   &lt;dbl&gt;       &lt;int&gt; &lt;int&gt;
1      1         653   405
2      2        1163   741
3      3         474   373
4      4         821   526
5      5         761   576
6      6         777   534</code></pre>
</div>
</div>
<p><strong>Network Statistics:</strong></p>
<p><strong>Total nodes:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gorder</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7231</code></pre>
</div>
</div>
<p><strong>Total edges:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gsize</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1379925</code></pre>
</div>
</div>
<p><strong>Seed nodes:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">V</span>(g)<span class="sc">$</span>seed)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2073</code></pre>
</div>
</div>
<p><strong>Modules:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>module))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12</code></pre>
</div>
</div>
<p><strong>Density:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">edge_density</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.05278965</code></pre>
</div>
</div>
<p><strong>Transitivity:</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">transitivity</span>(g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.44486</code></pre>
</div>
</div>
</section>
<section id="over-representation-analysis-ora" class="level2">
<h2 class="anchored" data-anchor-id="over-representation-analysis-ora">Over-Representation Analysis (ORA)</h2>
<p>To biologically interpret the network modules, we perform Reactome over-representation analysis. For each module, the most significantly enriched pathway is used as a representative label. This links graph-based communities to biological processes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>reactome_t2g <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(</span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">species =</span> <span class="st">"Homo sapiens"</span>,</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">category =</span> <span class="st">"C2"</span>,</span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">subcategory =</span> <span class="st">"REACTOME"</span></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">term =</span> gs_name,</span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene =</span> entrez_gene</span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(reactome_t2g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  2
  term                             gene
  &lt;chr&gt;                           &lt;int&gt;
1 REACTOME_2_LTR_CIRCLE_FORMATION  8815
2 REACTOME_2_LTR_CIRCLE_FORMATION  3159
3 REACTOME_2_LTR_CIRCLE_FORMATION  3981
4 REACTOME_2_LTR_CIRCLE_FORMATION 11168
5 REACTOME_2_LTR_CIRCLE_FORMATION  7518
6 REACTOME_2_LTR_CIRCLE_FORMATION  7520</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>module))</span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>ora_results <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">"list"</span>, <span class="fu">length</span>(mods))</span>
<span id="cb128-3"><a href="#cb128-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ora_results) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Module_"</span>, mods)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(mods)) {</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">&lt;-</span> mods[i]</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>  mod_genes <span class="ot">&lt;-</span> nodes_out<span class="sc">$</span>symbol[</span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>    nodes_out<span class="sc">$</span>module <span class="sc">==</span> m</span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  ]</span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">enrich_module</span>(</span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_symbol =</span> mod_genes,</span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">universe_symbol =</span> universe_symbols,</span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag =</span> tag,</span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">mod_id =</span> m,</span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">output_dir =</span> <span class="st">"."</span></span>
<span id="cb129-14"><a href="#cb129-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb129-15"><a href="#cb129-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-16"><a href="#cb129-16" aria-hidden="true" tabindex="-1"></a>  ora_results[[i]] <span class="ot">&lt;-</span> res</span>
<span id="cb129-17"><a href="#cb129-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb129-18"><a href="#cb129-18" aria-hidden="true" tabindex="-1"></a>ora_results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$Module_1
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:653] "1" "10057" "30" "176" "1636" "55331" "47" "10120" "100" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...62 enriched terms found
'data.frame':   62 obs. of  12 variables:
 $ ID            : chr  "REACTOME_NEUTROPHIL_DEGRANULATION" "REACTOME_GLYCOSAMINOGLYCAN_METABOLISM" "REACTOME_METABOLISM_OF_CARBOHYDRATES" "REACTOME_DISEASES_OF_METABOLISM" ...
 $ Description   : chr  "REACTOME_NEUTROPHIL_DEGRANULATION" "REACTOME_GLYCOSAMINOGLYCAN_METABOLISM" "REACTOME_METABOLISM_OF_CARBOHYDRATES" "REACTOME_DISEASES_OF_METABOLISM" ...
 $ GeneRatio     : chr  "263/653" "86/653" "127/653" "96/653" ...
 $ BgRatio       : chr  "401/7857" "104/7857" "245/7857" "191/7857" ...
 $ RichFactor    : num  0.656 0.827 0.518 0.503 0.647 ...
 $ FoldEnrichment: num  7.89 9.95 6.24 6.05 7.78 ...
 $ zScore        : num  42.6 27.7 25.1 21.3 22.1 ...
 $ pvalue        : num  3.50e-202 9.66e-77 2.07e-74 3.52e-54 2.89e-53 ...
 $ p.adjust      : num  1.26e-199 1.74e-74 2.50e-72 3.18e-52 2.09e-51 ...
 $ qvalue        : num  1.06e-199 1.46e-74 2.10e-72 2.67e-52 1.75e-51 ...
 $ geneID        : chr  "1/30/47/10120/102/101/175/178/10555/210/221/226/230/272/196527/290/327/353/127829/25852/410/427/83858/23250/232"| __truncated__ "10057/176/375790/126792/26229/10678/10331/79369/93010/2683/11285/63827/633/960/57214/11261/79586/54480/8534/505"| __truncated__ "10057/176/178/375790/226/230/126792/26229/10678/10331/79369/93010/2683/11285/63827/633/960/57214/11261/79586/54"| __truncated__ "176/9510/81794/81792/140766/170689/170690/170691/170692/9509/9508/11096/11174/56999/92949/375790/126792/26229/2"| __truncated__ ...
 $ Count         : int  263 86 127 96 75 60 51 42 56 43 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_2
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:1163] "2" "14" "18" "10257" "6833" "6059" "26090" "57406" "2532" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...352 enriched terms found
'data.frame':   352 obs. of  12 variables:
 $ ID            : chr  "REACTOME_SIGNALING_BY_GPCR" "REACTOME_SIGNALING_BY_INTERLEUKINS" "REACTOME_GPCR_LIGAND_BINDING" "REACTOME_G_ALPHA_I_SIGNALLING_EVENTS" ...
 $ Description   : chr  "REACTOME_SIGNALING_BY_GPCR" "REACTOME_SIGNALING_BY_INTERLEUKINS" "REACTOME_GPCR_LIGAND_BINDING" "REACTOME_G_ALPHA_I_SIGNALLING_EVENTS" ...
 $ GeneRatio     : chr  "290/1163" "279/1163" "178/1163" "160/1163" ...
 $ BgRatio       : chr  "376/7857" "369/7857" "188/7857" "172/7857" ...
 $ RichFactor    : num  0.771 0.756 0.947 0.93 0.971 ...
 $ FoldEnrichment: num  5.21 5.11 6.4 6.28 6.56 ...
 $ zScore        : num  34.9 33.7 31.2 29.2 27.4 ...
 $ pvalue        : num  3.24e-174 5.51e-163 2.96e-138 1.69e-120 4.28e-107 ...
 $ p.adjust      : num  2.62e-171 2.22e-160 7.96e-136 3.41e-118 6.91e-105 ...
 $ qvalue        : num  1.49e-171 1.27e-160 4.53e-136 1.95e-118 3.94e-105 ...
 $ geneID        : chr  "26090/57406/2532/57007/51554/107/108/109/196883/112/113/115/133/134/136/140/150/152/153/183/10768/207/301/8862/"| __truncated__ "9049/207/240/301/302/351/466/1386/10538/596/598/29760/54014/834/836/867/6356/6363/6347/6364/6367/6351/6352/1230"| __truncated__ "2532/57007/51554/133/134/136/140/150/152/153/183/301/8862/187/351/552/624/718/719/727/728/10203/846/887/6356/63"| __truncated__ "57007/107/108/109/196883/112/113/115/134/140/150/152/183/10768/301/8862/187/351/624/718/719/727/728/801/817/818"| __truncated__ ...
 $ Count         : int  290 279 178 160 133 167 157 242 109 86 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_3
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:474] "8086" "23536" "134637" "113179" "83440" "25909" "54806" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...140 enriched terms found
'data.frame':   140 obs. of  12 variables:
 $ ID            : chr  "REACTOME_CELL_CYCLE_MITOTIC" "REACTOME_M_PHASE" "REACTOME_MITOTIC_PROMETAPHASE" "REACTOME_CILIUM_ASSEMBLY" ...
 $ Description   : chr  "REACTOME_CELL_CYCLE_MITOTIC" "REACTOME_M_PHASE" "REACTOME_MITOTIC_PROMETAPHASE" "REACTOME_CILIUM_ASSEMBLY" ...
 $ GeneRatio     : chr  "296/474" "223/474" "150/474" "126/474" ...
 $ BgRatio       : chr  "491/7857" "353/7857" "186/7857" "177/7857" ...
 $ RichFactor    : num  0.603 0.632 0.806 0.712 0.595 ...
 $ FoldEnrichment: num  9.99 10.47 13.37 11.8 9.87 ...
 $ zScore        : num  52.1 46.1 43.3 36.8 33.4 ...
 $ pvalue        : num  4.41e-272 4.55e-201 1.16e-156 6.22e-118 1.47e-104 ...
 $ p.adjust      : num  1.98e-269 1.02e-198 1.73e-154 7.00e-116 1.32e-102 ...
 $ qvalue        : num  1.40e-269 7.21e-199 1.22e-154 4.93e-116 9.29e-103 ...
 $ geneID        : chr  "8086/25909/84962/10142/7840/10393/51529/25906/119504/29882/29945/51433/51434/23141/10776/9212/8815/332/8548/798"| __truncated__ "8086/25909/10142/7840/10393/51529/25906/119504/29882/29945/51433/51434/23141/10776/9212/8815/332/8548/699/701/9"| __truncated__ "25909/10142/7840/9212/332/699/701/9184/891/9133/9738/991/113130/55143/983/55755/1058/1060/1063/64946/2491/55835"| __truncated__ "54806/10142/7840/403/84100/50807/79969/27077/92482/582/79738/166379/583/585/55212/27241/26005/57545/9738/10575/"| __truncated__ ...
 $ Count         : int  296 223 150 126 128 89 84 126 85 100 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_4
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:821] "65985" "51166" "60496" "10157" "4363" "8714" "215" "5826" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...96 enriched terms found
'data.frame':   96 obs. of  12 variables:
 $ ID            : chr  "REACTOME_SLC_MEDIATED_TRANSMEMBRANE_TRANSPORT" "REACTOME_FATTY_ACID_METABOLISM" "REACTOME_BIOLOGICAL_OXIDATIONS" "REACTOME_PHOSPHOLIPID_METABOLISM" ...
 $ Description   : chr  "REACTOME_SLC_MEDIATED_TRANSMEMBRANE_TRANSPORT" "REACTOME_FATTY_ACID_METABOLISM" "REACTOME_BIOLOGICAL_OXIDATIONS" "REACTOME_PHOSPHOLIPID_METABOLISM" ...
 $ GeneRatio     : chr  "151/821" "111/821" "109/821" "116/821" ...
 $ BgRatio       : chr  "160/7857" "133/7857" "131/7857" "172/7857" ...
 $ RichFactor    : num  0.944 0.835 0.832 0.674 0.838 ...
 $ FoldEnrichment: num  9.03 7.99 7.96 6.45 8.02 ...
 $ zScore        : num  35.1 27.8 27.5 24.7 24.7 ...
 $ pvalue        : num  8.75e-141 1.09e-88 8.93e-87 3.28e-74 1.91e-70 ...
 $ p.adjust      : num  2.19e-138 1.36e-86 7.44e-85 2.05e-72 9.57e-69 ...
 $ qvalue        : num  1.40e-138 8.71e-87 4.76e-85 1.31e-72 6.12e-69 ...
 $ geneID        : chr  "23382/347/402/23568/682/1356/1497/133418/9843/286256/51248/57127/4891/6557/6558/6560/9990/10723/9058/64849/6564"| __truncated__ "4363/215/10449/31/32/80724/33/34/35/37/79777/84320/26027/55856/122970/11332/10005/23597/51/8309/8310/80221/2180"| __truncated__ "55347/84836/54988/84532/55902/125/126/127/128/196/10327/8574/22977/216/219/8639/405/9915/57412/670/10380/874/88"| __truncated__ "171586/63874/51205/55750/10554/56895/55326/10423/1040/8760/10390/1119/1120/56994/27132/54675/80821/23259/84649/"| __truncated__ ...
 $ Count         : int  151 111 109 116 88 102 68 98 71 55 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_5
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:761] "22848" "64746" "59272" "60" "55860" "10121" "10097" "10096" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...134 enriched terms found
'data.frame':   134 obs. of  12 variables:
 $ ID            : chr  "REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION" "REACTOME_INTRA_GOLGI_AND_RETROGRADE_GOLGI_TO_ER_TRAFFIC" "REACTOME_TRANSPORT_TO_THE_GOLGI_AND_SUBSEQUENT_MODIFICATION" "REACTOME_ER_TO_GOLGI_ANTEROGRADE_TRANSPORT" ...
 $ Description   : chr  "REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION" "REACTOME_INTRA_GOLGI_AND_RETROGRADE_GOLGI_TO_ER_TRAFFIC" "REACTOME_TRANSPORT_TO_THE_GOLGI_AND_SUBSEQUENT_MODIFICATION" "REACTOME_ER_TO_GOLGI_ANTEROGRADE_TRANSPORT" ...
 $ GeneRatio     : chr  "239/761" "175/761" "157/761" "134/761" ...
 $ BgRatio       : chr  "266/7857" "179/7857" "162/7857" "136/7857" ...
 $ RichFactor    : num  0.898 0.978 0.969 0.985 0.983 ...
 $ FoldEnrichment: num  9.28 10.09 10.01 10.17 10.15 ...
 $ zScore        : num  45 40.3 37.9 35.3 32.5 ...
 $ pvalue        : num  1.07e-223 2.87e-179 5.55e-158 1.30e-137 4.49e-116 ...
 $ p.adjust      : num  5.39e-221 7.25e-177 9.35e-156 1.64e-135 4.53e-114 ...
 $ qvalue        : num  4.01e-221 5.40e-177 6.96e-156 1.22e-135 3.37e-114 ...
 $ geneID        : chr  "55860/10121/56052/84920/79087/79868/199857/85365/10195/29880/29929/79053/79796/267/287/288/23243/372/374/375/37"| __truncated__ "55860/10121/56894/372/375/377/378/381/55738/84364/26286/23647/10139/400/51272/636/23299/662/829/830/832/1062/93"| __truncated__ "55860/10121/287/288/23243/372/374/375/377/378/381/55738/84364/26286/8704/8703/8702/9334/9331/10282/51272/829/83"| __truncated__ "55860/10121/287/288/23243/372/374/375/377/378/381/55738/84364/26286/10282/51272/829/830/832/1604/966/10175/2542"| __truncated__ ...
 $ Count         : int  239 175 157 134 114 119 106 86 82 82 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_6
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:777] "57505" "3983" "84448" "22885" "48" "72" "8038" "8751" "8754" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...105 enriched terms found
'data.frame':   105 obs. of  12 variables:
 $ ID            : chr  "REACTOME_RRNA_PROCESSING" "REACTOME_NERVOUS_SYSTEM_DEVELOPMENT" "REACTOME_EUKARYOTIC_TRANSLATION_INITIATION" "REACTOME_EXTRACELLULAR_MATRIX_ORGANIZATION" ...
 $ Description   : chr  "REACTOME_RRNA_PROCESSING" "REACTOME_NERVOUS_SYSTEM_DEVELOPMENT" "REACTOME_EUKARYOTIC_TRANSLATION_INITIATION" "REACTOME_EXTRACELLULAR_MATRIX_ORGANIZATION" ...
 $ GeneRatio     : chr  "158/777" "245/777" "91/777" "146/777" ...
 $ BgRatio       : chr  "170/7857" "467/7857" "94/7857" "244/7857" ...
 $ RichFactor    : num  0.929 0.525 0.968 0.598 0.921 ...
 $ FoldEnrichment: num  9.4 5.3 9.79 6.05 9.32 ...
 $ zScore        : num  36.7 31.8 28.4 26.5 26.1 ...
 $ pvalue        : num  8.90e-149 3.35e-132 2.58e-89 1.07e-86 2.63e-75 ...
 $ p.adjust      : num  3.38e-146 6.37e-130 3.27e-87 1.02e-84 2.00e-73 ...
 $ qvalue        : num  2.48e-146 4.67e-130 2.40e-87 7.49e-85 1.47e-73 ...
 $ geneID        : chr  "9790/705/10438/25879/9188/54555/11056/57647/27292/22894/60528/10436/90459/51013/5394/23404/51010/54512/56915/11"| __truncated__ "3983/84448/22885/116986/9495/214/51107/83464/10095/9048/775/776/8913/8912/8911/782/784/10486/100133941/971/1075"| __truncated__ "1964/1967/8892/8891/8890/8893/1965/8894/1968/8661/8662/8663/8664/3646/8665/8666/8667/8668/8669/27335/51386/1048"| __truncated__ "8038/8751/8754/54829/649/652/655/11132/147968/92291/6650/824/825/827/23473/388743/10753/826/831/977/999/1300/13"| __truncated__ ...
 $ Count         : int  158 245 91 146 82 99 136 91 73 78 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_7
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:461] "26574" "5825" "10006" "10152" "27" "29" "91452" "58" "87" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...75 enriched terms found
'data.frame':   75 obs. of  12 variables:
 $ ID            : chr  "REACTOME_RHO_GTPASE_CYCLE" "REACTOME_RAC1_GTPASE_CYCLE" "REACTOME_CDC42_GTPASE_CYCLE" "REACTOME_RHOA_GTPASE_CYCLE" ...
 $ Description   : chr  "REACTOME_RHO_GTPASE_CYCLE" "REACTOME_RAC1_GTPASE_CYCLE" "REACTOME_CDC42_GTPASE_CYCLE" "REACTOME_RHOA_GTPASE_CYCLE" ...
 $ GeneRatio     : chr  "363/461" "160/461" "135/461" "123/461" ...
 $ BgRatio       : chr  "419/7857" "172/7857" "144/7857" "136/7857" ...
 $ RichFactor    : num  0.866 0.93 0.938 0.904 0.895 ...
 $ FoldEnrichment: num  14.8 15.9 16 15.4 15.3 ...
 $ zScore        : num  72.3 49.2 45.3 42.3 33.2 ...
 $ pvalue        : num  0.00 5.74e-193 2.15e-162 3.77e-142 7.53e-87 ...
 $ p.adjust      : num  0.00 1.25e-190 3.12e-160 4.10e-140 6.55e-85 ...
 $ qvalue        : num  0.00 1.04e-190 2.60e-160 3.41e-140 5.45e-85 ...
 $ geneID        : chr  "5825/10006/10152/27/29/91452/87/120/9590/11214/224/347902/51479/22852/54443/116985/64411/392/79658/94134/55114/"| __truncated__ "10006/10152/27/29/347902/116985/64411/392/79658/94134/55114/57584/58504/57636/83478/9938/23092/201176/9411/2571"| __truncated__ "29/116985/64411/392/79658/55114/57584/58504/83478/23092/201176/9411/257106/57514/9743/115703/2909/80728/393/143"| __truncated__ "5825/29/91452/11214/54443/116985/64411/392/79658/93663/84986/57584/58504/57636/83478/23092/79822/9411/257106/57"| __truncated__ ...
 $ Count         : int  363 160 135 123 77 76 63 59 62 54 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_8
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:1111] "19" "25" "28976" "65057" "86" "9370" "10974" "121536" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...223 enriched terms found
'data.frame':   223 obs. of  12 variables:
 $ ID            : chr  "REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53" "REACTOME_DNA_REPAIR" "REACTOME_CHROMATIN_MODIFYING_ENZYMES" "REACTOME_REGULATION_OF_TP53_ACTIVITY" ...
 $ Description   : chr  "REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53" "REACTOME_DNA_REPAIR" "REACTOME_CHROMATIN_MODIFYING_ENZYMES" "REACTOME_REGULATION_OF_TP53_ACTIVITY" ...
 $ GeneRatio     : chr  "288/1111" "197/1111" "158/1111" "125/1111" ...
 $ BgRatio       : chr  "342/7857" "285/7857" "216/7857" "149/7857" ...
 $ RichFactor    : num  0.842 0.691 0.731 0.839 0.78 ...
 $ FoldEnrichment: num  5.96 4.89 5.17 5.93 5.52 ...
 $ zScore        : num  38 27.1 25.2 24.7 21.2 ...
 $ pvalue        : num  5.44e-200 1.09e-104 2.57e-89 1.55e-83 1.09e-62 ...
 $ p.adjust      : num  3.04e-197 3.04e-102 4.79e-87 2.16e-81 1.21e-60 ...
 $ qvalue        : num  1.88e-197 1.88e-102 2.96e-87 1.34e-81 7.51e-61 ...
 $ geneID        : chr  "26523/27161/192669/192670/84883/208/10000/317/1820/472/545/6790/54971/580/581/79370/604/637/641/665/672/23774/2"| __truncated__ "25/65057/86/328/472/545/29086/8314/580/9031/641/672/675/79184/83990/8900/890/902/1017/1022/9557/1111/11200/8451"| __truncated__ "86/121536/8289/57492/196528/5926/51742/84159/55870/9070/55729/56970/23774/10902/25855/7862/27154/10498/595/1019"| __truncated__ "208/10000/472/545/6790/54971/580/641/672/23774/29117/83990/7862/27154/8900/890/900/1017/1020/8851/1029/1107/110"| __truncated__ ...
 $ Count         : int  288 197 158 125 103 84 88 83 84 106 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_9
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:740] "26154" "20" "21" "24" "23461" "23460" "10347" "10351" "10350" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...190 enriched terms found
'data.frame':   190 obs. of  12 variables:
 $ ID            : chr  "REACTOME_CLASS_I_MHC_MEDIATED_ANTIGEN_PROCESSING_PRESENTATION" "REACTOME_ANTIGEN_PROCESSING_UBIQUITINATION_PROTEASOME_DEGRADATION" "REACTOME_NEDDYLATION" "REACTOME_DEUBIQUITINATION" ...
 $ Description   : chr  "REACTOME_CLASS_I_MHC_MEDIATED_ANTIGEN_PROCESSING_PRESENTATION" "REACTOME_ANTIGEN_PROCESSING_UBIQUITINATION_PROTEASOME_DEGRADATION" "REACTOME_NEDDYLATION" "REACTOME_DEUBIQUITINATION" ...
 $ GeneRatio     : chr  "271/740" "243/740" "170/740" "149/740" ...
 $ BgRatio       : chr  "335/7857" "269/7857" "204/7857" "236/7857" ...
 $ RichFactor    : num  0.809 0.903 0.833 0.631 0.655 ...
 $ FoldEnrichment: num  8.59 9.59 8.85 6.7 6.96 ...
 $ zScore        : num  45.8 46.2 36.6 28.7 25.6 ...
 $ pvalue        : num  1.65e-233 1.14e-232 7.48e-146 9.27e-97 4.99e-76 ...
 $ p.adjust      : num  8.61e-231 2.97e-230 1.30e-143 1.21e-94 5.22e-74 ...
 $ qvalue        : num  5.44e-231 1.88e-230 8.24e-144 7.66e-95 3.30e-74 ...
 $ geneID        : chr  "25847/9870/10425/51665/79754/140459/140460/140461/140462/10533/642/53339/90135/8945/821/868/899/997/1514/1515/8"| __truncated__ "25847/9870/10425/51665/79754/140459/140460/140461/140462/10533/642/53339/90135/8945/868/899/997/8454/8453/8452/"| __truncated__ "122416/51665/79754/140459/140460/140461/140462/53339/90135/8945/28952/83987/899/150684/51397/51122/54939/28991/"| __truncated__ "79913/93973/11047/324/171023/55252/6314/8312/8313/8678/1540/23770/221937/3607/3054/3416/64135/54617/83444/12547"| __truncated__ ...
 $ Count         : int  271 243 170 149 114 109 131 73 72 68 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_11
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:173] "10189" "9716" "10286" "8896" "55749" "10421" "51362" "988" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...17 enriched terms found
'data.frame':   17 obs. of  12 variables:
 $ ID            : chr  "REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA" "REACTOME_MRNA_SPLICING" "REACTOME_RNA_POLYMERASE_II_TRANSCRIPTION_TERMINATION" "REACTOME_MRNA_SPLICING_MINOR_PATHWAY" ...
 $ Description   : chr  "REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA" "REACTOME_MRNA_SPLICING" "REACTOME_RNA_POLYMERASE_II_TRANSCRIPTION_TERMINATION" "REACTOME_MRNA_SPLICING_MINOR_PATHWAY" ...
 $ GeneRatio     : chr  "169/173" "151/173" "49/173" "36/173" ...
 $ BgRatio       : chr  "225/7857" "177/7857" "56/7857" "51/7857" ...
 $ RichFactor    : num  0.751 0.853 0.875 0.706 0.5 ...
 $ FoldEnrichment: num  34.1 38.7 39.7 32.1 22.7 ...
 $ zScore        : num  75.6 76.2 43.7 33.4 27.8 ...
 $ pvalue        : num  1.92e-292 8.58e-265 8.03e-77 1.16e-49 1.12e-41 ...
 $ p.adjust      : num  1.89e-290 4.21e-263 2.62e-75 2.84e-48 2.20e-40 ...
 $ qvalue        : num  1.58e-290 3.52e-263 2.20e-75 2.38e-48 1.84e-40 ...
 $ geneID        : chr  "10189/9716/10286/8896/55749/10421/51362/988/10523/26097/10978/53981/51692/10898/51340/1477/1478/23283/1479/5625"| __truncated__ "10189/9716/10286/8896/55749/10421/51362/988/10523/10978/53981/51692/10898/51340/1477/1478/23283/1479/56259/5150"| __truncated__ "10189/51362/26097/10978/53981/51692/10898/1477/1478/23283/1479/10212/9785/81608/84248/84967/134353/4686/22916/1"| __truncated__ "9416/11325/9343/57819/4686/22916/10081/24148/10594/55599/23451/10992/23450/10262/83443/51639/23020/79622/11066/"| __truncated__ ...
 $ Count         : int  169 151 49 36 36 25 16 11 11 10 ...
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_13
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:91] "54998" "118487" "7818" "26284" "90480" "85476" "84340" "65008" ...
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...2 enriched terms found
'data.frame':   2 obs. of  12 variables:
 $ ID            : chr  "REACTOME_MITOCHONDRIAL_TRANSLATION" "REACTOME_TRANSLATION"
 $ Description   : chr  "REACTOME_MITOCHONDRIAL_TRANSLATION" "REACTOME_TRANSLATION"
 $ GeneRatio     : chr  "91/91" "91/91"
 $ BgRatio       : chr  "92/7857" "259/7857"
 $ RichFactor    : num  0.989 0.351
 $ FoldEnrichment: num  85.4 30.3
 $ zScore        : num  88.1 52
 $ pvalue        : num  7.14e-213 3.35e-143
 $ p.adjust      : num  5.00e-212 1.17e-142
 $ qvalue        : num  3.76e-212 8.81e-143
 $ geneID        : chr  "54998/118487/7818/26284/90480/85476/84340/65008/124995/65003/6182/28998/64928/29088/54948/63875/29074/9801/5106"| __truncated__ "54998/118487/7818/26284/90480/85476/84340/65008/124995/65003/6182/28998/64928/29088/54948/63875/29074/9801/5106"| __truncated__
 $ Count         : int  91 91
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 


$Module_14
#
# over-representation test
#
#...@organism    UNKNOWN 
#...@ontology    UNKNOWN 
#...@gene    chr [1:6] "9807" "64768" "55190" "11165" "9677" "23262"
#...pvalues adjusted by 'BH' with cutoff &lt;0.05 
#...1 enriched terms found
'data.frame':   1 obs. of  12 variables:
 $ ID            : chr "REACTOME_INOSITOL_PHOSPHATE_METABOLISM"
 $ Description   : chr "REACTOME_INOSITOL_PHOSPHATE_METABOLISM"
 $ GeneRatio     : chr "6/6"
 $ BgRatio       : chr "38/7857"
 $ RichFactor    : num 0.158
 $ FoldEnrichment: num 207
 $ zScore        : num 35.1
 $ pvalue        : num 8.47e-15
 $ p.adjust      : num 8.47e-15
 $ qvalue        : logi NA
 $ geneID        : chr "9807/64768/55190/11165/9677/23262"
 $ Count         : int 6
#...Citation
S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang, W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize multiomics data. Nature Protocols. 2024, 19(11):3292-3320 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>module_enrich_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">seq_along</span>(ora_results),</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i) {</span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> ora_results[[i]]</span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(res)) {</span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-13"><a href="#cb131-13" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">nrow</span>(df) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb131-14"><a href="#cb131-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb131-15"><a href="#cb131-15" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb131-16"><a href="#cb131-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-17"><a href="#cb131-17" aria-hidden="true" tabindex="-1"></a>      df<span class="sc">$</span>module <span class="ot">&lt;-</span> mods[i]</span>
<span id="cb131-18"><a href="#cb131-18" aria-hidden="true" tabindex="-1"></a>      df</span>
<span id="cb131-19"><a href="#cb131-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb131-20"><a href="#cb131-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb131-21"><a href="#cb131-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb131-22"><a href="#cb131-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-23"><a href="#cb131-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(module_enrich_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                            ID
REACTOME_NEUTROPHIL_DEGRANULATION            REACTOME_NEUTROPHIL_DEGRANULATION
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM    REACTOME_GLYCOSAMINOGLYCAN_METABOLISM
REACTOME_METABOLISM_OF_CARBOHYDRATES...3  REACTOME_METABOLISM_OF_CARBOHYDRATES
REACTOME_DISEASES_OF_METABOLISM...4            REACTOME_DISEASES_OF_METABOLISM
REACTOME_DISEASES_OF_GLYCOSYLATION...5      REACTOME_DISEASES_OF_GLYCOSYLATION
REACTOME_SPHINGOLIPID_METABOLISM              REACTOME_SPHINGOLIPID_METABOLISM
                                                                   Description
REACTOME_NEUTROPHIL_DEGRANULATION            REACTOME_NEUTROPHIL_DEGRANULATION
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM    REACTOME_GLYCOSAMINOGLYCAN_METABOLISM
REACTOME_METABOLISM_OF_CARBOHYDRATES...3  REACTOME_METABOLISM_OF_CARBOHYDRATES
REACTOME_DISEASES_OF_METABOLISM...4            REACTOME_DISEASES_OF_METABOLISM
REACTOME_DISEASES_OF_GLYCOSYLATION...5      REACTOME_DISEASES_OF_GLYCOSYLATION
REACTOME_SPHINGOLIPID_METABOLISM              REACTOME_SPHINGOLIPID_METABOLISM
                                         GeneRatio  BgRatio RichFactor
REACTOME_NEUTROPHIL_DEGRANULATION          263/653 401/7857  0.6558603
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM       86/653 104/7857  0.8269231
REACTOME_METABOLISM_OF_CARBOHYDRATES...3   127/653 245/7857  0.5183673
REACTOME_DISEASES_OF_METABOLISM...4         96/653 191/7857  0.5026178
REACTOME_DISEASES_OF_GLYCOSYLATION...5      75/653 116/7857  0.6465517
REACTOME_SPHINGOLIPID_METABOLISM            60/653  73/7857  0.8219178
                                         FoldEnrichment   zScore        pvalue
REACTOME_NEUTROPHIL_DEGRANULATION              7.891416 42.64795 3.497451e-202
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM          9.949670 27.66045  9.662526e-77
REACTOME_METABOLISM_OF_CARBOHYDRATES...3       6.237078 25.07224  2.073865e-74
REACTOME_DISEASES_OF_METABOLISM...4            6.047577 21.26110  3.519974e-54
REACTOME_DISEASES_OF_GLYCOSYLATION...5         7.779413 22.14589  2.890547e-53
REACTOME_SPHINGOLIPID_METABOLISM               9.889446 22.97234  3.625769e-53
                                              p.adjust        qvalue
REACTOME_NEUTROPHIL_DEGRANULATION        1.262580e-199 1.060280e-199
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM     1.744086e-74  1.464636e-74
REACTOME_METABOLISM_OF_CARBOHYDRATES...3  2.495551e-72  2.095695e-72
REACTOME_DISEASES_OF_METABOLISM...4       3.176777e-52  2.667770e-52
REACTOME_DISEASES_OF_GLYCOSYLATION...5    2.086975e-51  1.752584e-51
REACTOME_SPHINGOLIPID_METABOLISM          2.181505e-51  1.831968e-51
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            geneID
REACTOME_NEUTROPHIL_DEGRANULATION        1/30/47/10120/102/101/175/178/10555/210/221/226/230/272/196527/290/327/353/127829/25852/410/427/83858/23250/23200/10159/535/527/51382/10396/567/2683/51411/25798/683/684/79169/387263/51719/51806/820/55832/124583/10487/823/847/10694/929/11314/960/961/963/965/967/22918/8621/634/1084/4680/1675/1116/1118/160364/23601/54918/23412/29099/23406/8904/8895/55313/8804/10321/83716/1471/1476/5476/1508/1509/1512/1520/1727/84418/1650/1654/8560/51071/8694/23317/5611/80331/1774/79930/29952/6993/1915/1938/22875/23071/2171/2207/2212/2215/2219/10875/2268/2444/2517/2519/2548/2588/25801/8836/84514/2717/11010/2760/9535/51125/2821/53831/2896/2934/2950/2990/2992/3043/23593/3074/138050/3101/3106/3107/3240/10855/3310/84329/3417/3608/3614/3615/3683/3685/56888/8514/3903/3916/3920/3958/10288/11025/79888/116844/4057/4069/84061/4125/4126/25840/8972/4257/9761/4311/64386/4332/9961/93100/23218/23385/4758/150372/23114/387921/56954/10577/10562/4973/5004/94103/26578/5036/11240/11333/8566/5175/5211/5223/5236/55276/10857/5284/5315/5317/51316/5328/5329/80301/4860/5547/10549/9588/5645/5646/5663/80142/5787/5788/5795/5799/5806/29108/5834/5836/25797/5768/53917/9545/5912/57826/8635/6282/6278/6279/6280/6286/9522/6386/6402/12/1992/6317/5269/140885/6556/121260/11001/6515/6518/57153/6590/6727/114790/258010/9145/10312/6947/7077/7097/11322/55754/9725/7130/10043/7226/81619/7305/201294/9218/10493/8876/7520/2547/51646
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    10057/176/375790/126792/26229/10678/10331/79369/93010/2683/11285/63827/633/960/57214/11261/79586/54480/8534/50515/55501/113189/51363/9435/9469/56548/83539/22856/337876/55790/55454/1464/10675/1634/29940/92126/2131/2132/2331/2588/79411/26035/2817/221914/2719/2239/10082/2990/3037/3038/3073/3074/138050/10855/9653/9957/9956/9955/9953/9394/90161/266722/3339/8692/3423/3425/11081/4060/10894/4669/1463/3340/8509/4969/4958/9061/9060/5549/6383/9672/6385/6448/10090/1462/64131/64132
REACTOME_METABOLISM_OF_CARBOHYDRATES...3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               10057/176/178/375790/226/230/126792/26229/10678/10331/79369/93010/2683/11285/63827/633/960/57214/11261/79586/54480/8534/50515/55501/113189/51363/9435/9469/56548/83539/22856/337876/55790/55454/1464/10675/1634/51181/51071/29940/92126/7957/2131/2132/2331/92579/2548/2582/2588/2592/2632/79411/26035/2817/221914/2719/2239/10082/2821/2990/2992/8908/2997/3037/3038/3073/3074/138050/3101/10855/9653/9957/9956/9955/9953/9394/90161/266722/3339/8692/3423/3425/11081/3795/4060/10894/4125/4126/4669/1463/3340/8509/4969/4958/9061/9060/5211/5223/5226/25796/5236/55276/5255/5256/5257/5261/5315/5507/5549/5631/5634/5834/5836/64080/6120/22934/6383/9672/6385/6448/23729/2542/7086/10090/1462/64131/64132
REACTOME_DISEASES_OF_METABOLISM...4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         176/9510/81794/81792/140766/170689/170690/170691/170692/9509/9508/11096/11174/56999/92949/375790/126792/26229/2683/11285/63827/633/56913/29071/113189/9469/22856/1464/10675/5476/1634/51181/7957/2131/2132/2331/92579/2548/2582/2588/79695/2591/2592/2632/2817/221914/2719/2239/10082/2990/2992/8908/2997/3073/3074/138050/3339/3417/3423/3425/11081/3795/3938/3955/4060/143662/94025/200958/4585/727897/4588/118430/4669/1463/4758/4969/4958/9060/5236/55624/10585/29954/5507/5549/22934/157869/6383/9672/6385/6448/2542/10418/7058/79875/221981/1462
REACTOME_DISEASES_OF_GLYCOSYLATION...5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    176/9510/81794/81792/140766/170689/170690/170691/170692/9509/9508/11096/11174/56999/92949/375790/126792/26229/2683/11285/63827/633/56913/29071/113189/9469/22856/1464/10675/5476/1634/2131/2132/2331/2582/79695/2591/2592/2817/221914/2719/2239/10082/3073/3074/3339/11081/3955/4060/143662/94025/200958/4585/727897/4588/118430/1463/4758/4969/4958/9060/5236/55624/10585/29954/5549/157869/6383/9672/6385/10418/7058/79875/221981/1462
REACTOME_SPHINGOLIPID_METABOLISM                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            55331/221/222/410/414/22901/79642/153642/427/8706/64781/29956/79603/91012/253782/80772/5476/8560/123099/339221/57488/79152/2581/57704/2717/79411/51228/2760/3073/3074/2531/4758/94101/29095/94103/151742/5587/25865/23683/142891/259230/166929/8879/81537/130367/6609/6610/55512/124976/10558/9517/55304/171546/165679/412/285362/25870/7357/7368/9218
                                         Count module
REACTOME_NEUTROPHIL_DEGRANULATION          263      1
REACTOME_GLYCOSAMINOGLYCAN_METABOLISM       86      1
REACTOME_METABOLISM_OF_CARBOHYDRATES...3   127      1
REACTOME_DISEASES_OF_METABOLISM...4         96      1
REACTOME_DISEASES_OF_GLYCOSYLATION...5      75      1
REACTOME_SPHINGOLIPID_METABOLISM            60      1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>module_summary <span class="ot">&lt;-</span> module_enrich_df <span class="sc">|&gt;</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(module) <span class="sc">|&gt;</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">n_sig_pathways =</span> <span class="fu">n</span>(),</span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_term =</span> ID[<span class="fu">which.min</span>(p.adjust)],</span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_q =</span> <span class="fu">min</span>(p.adjust),</span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>module_summary <span class="ot">&lt;-</span> module_summary <span class="sc">|&gt;</span></span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>    mod_DE_info,</span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"module"</span></span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb133-15"><a href="#cb133-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(top_q)</span>
<span id="cb133-16"><a href="#cb133-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-17"><a href="#cb133-17" aria-hidden="true" tabindex="-1"></a>module_summary <span class="ot">&lt;-</span> module_summary <span class="sc">|&gt;</span></span>
<span id="cb133-18"><a href="#cb133-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb133-19"><a href="#cb133-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">module_label =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb133-20"><a href="#cb133-20" aria-hidden="true" tabindex="-1"></a>      top_term,</span>
<span id="cb133-21"><a href="#cb133-21" aria-hidden="true" tabindex="-1"></a>      <span class="st">"^REACTOME_"</span>,</span>
<span id="cb133-22"><a href="#cb133-22" aria-hidden="true" tabindex="-1"></a>      <span class="st">""</span></span>
<span id="cb133-23"><a href="#cb133-23" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb133-24"><a href="#cb133-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">module_label =</span> <span class="fu">str_replace_all</span>(</span>
<span id="cb133-25"><a href="#cb133-25" aria-hidden="true" tabindex="-1"></a>      module_label,</span>
<span id="cb133-26"><a href="#cb133-26" aria-hidden="true" tabindex="-1"></a>      <span class="st">"_"</span>,</span>
<span id="cb133-27"><a href="#cb133-27" aria-hidden="true" tabindex="-1"></a>      <span class="st">" "</span></span>
<span id="cb133-28"><a href="#cb133-28" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb133-29"><a href="#cb133-29" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb133-30"><a href="#cb133-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-31"><a href="#cb133-31" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(module_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  7
  module n_sig_pathways top_term            top_q module_size  n_DE module_label
   &lt;dbl&gt;          &lt;int&gt; &lt;chr&gt;               &lt;dbl&gt;       &lt;int&gt; &lt;int&gt; &lt;chr&gt;       
1      7             75 REACTOME_RHO_G 0                 461   321 RHO GTPASE 
2     11             17 REACTOME_PROCE 1.89e-290         173   143 PROCESSING 
3      3            140 REACTOME_CELL_ 1.98e-269         474   373 CELL CYCLE 
4      9            190 REACTOME_CLASS 8.61e-231         740   533 CLASS I MHC
5      5            134 REACTOME_ASPAR 5.39e-221         761   576 ASPARAGINE 
6     13              2 REACTOME_MITOC 5.00e-212          91    66 MITOCHONDRI</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>key_modules <span class="ot">&lt;-</span> module_summary <span class="sc">|&gt;</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>    n_sig_pathways <span class="sc">&gt;=</span> <span class="dv">2</span>,</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>    top_q <span class="sc">&lt;</span> <span class="fl">0.05</span>,</span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>    n_DE <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(top_q)</span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>key_modules <span class="sc">|&gt;</span></span>
<span id="cb135-10"><a href="#cb135-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(</span>
<span id="cb135-11"><a href="#cb135-11" aria-hidden="true" tabindex="-1"></a>    module,</span>
<span id="cb135-12"><a href="#cb135-12" aria-hidden="true" tabindex="-1"></a>    module_size,</span>
<span id="cb135-13"><a href="#cb135-13" aria-hidden="true" tabindex="-1"></a>    n_sig_pathways,</span>
<span id="cb135-14"><a href="#cb135-14" aria-hidden="true" tabindex="-1"></a>    top_q</span>
<span id="cb135-15"><a href="#cb135-15" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11  4
   module module_size n_sig_pathways     top_q
    &lt;dbl&gt;       &lt;int&gt;          &lt;int&gt;     &lt;dbl&gt;
 1      7         461             75 0        
 2     11         173             17 1.89e-290
 3      3         474            140 1.98e-269
 4      9         740            190 8.61e-231
 5      5         761            134 5.39e-221
 6     13          91              2 5.00e-212
 7      1         653             62 1.26e-199
 8      8        1111            223 3.04e-197
 9      2        1163            352 2.62e-171
10      6         777            105 3.38e-146
11      4         821             96 2.19e-138</code></pre>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(key_modules)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  7
  module n_sig_pathways top_term            top_q module_size  n_DE module_label
   &lt;dbl&gt;          &lt;int&gt; &lt;chr&gt;               &lt;dbl&gt;       &lt;int&gt; &lt;int&gt; &lt;chr&gt;       
1      7             75 REACTOME_RHO_G 0                 461   321 RHO GTPASE 
2     11             17 REACTOME_PROCE 1.89e-290         173   143 PROCESSING 
3      3            140 REACTOME_CELL_ 1.98e-269         474   373 CELL CYCLE 
4      9            190 REACTOME_CLASS 8.61e-231         740   533 CLASS I MHC
5      5            134 REACTOME_ASPAR 5.39e-221         761   576 ASPARAGINE 
6     13              2 REACTOME_MITOC 5.00e-212          91    66 MITOCHONDRI</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>nodes_out <span class="ot">&lt;-</span> nodes_out <span class="sc">|&gt;</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>    module_summary <span class="sc">|&gt;</span></span>
<span id="cb139-4"><a href="#cb139-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(module, module_label),</span>
<span id="cb139-5"><a href="#cb139-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"module"</span></span>
<span id="cb139-6"><a href="#cb139-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb139-7"><a href="#cb139-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-8"><a href="#cb139-8" aria-hidden="true" tabindex="-1"></a>nodes_out <span class="ot">&lt;-</span> nodes_out <span class="sc">|&gt;</span></span>
<span id="cb139-9"><a href="#cb139-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb139-10"><a href="#cb139-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">in_key_module =</span> module <span class="sc">%in%</span> key_modules<span class="sc">$</span>module</span>
<span id="cb139-11"><a href="#cb139-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb139-12"><a href="#cb139-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(in_key_module, <span class="at">.after =</span> module)</span>
<span id="cb139-13"><a href="#cb139-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nodes_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  11
  symbol hub   seed  module in_key_module degree strength betweenness  logFC
  &lt;chr&gt;  &lt;lgl&gt; &lt;lgl&gt;  &lt;dbl&gt; &lt;lgl&gt;          &lt;dbl&gt;    &lt;dbl&gt;       &lt;dbl&gt;  &lt;dbl&gt;
1 A1BG   FALSE FALSE      1 TRUE             588     1392  0.0000380   0.103
2 A2M    FALSE FALSE      2 TRUE             409     1006  0.000360    1.41 
3 AAAS   TRUE  FALSE      3 TRUE            1902     7631  0.00561    -0.177
4 AACS   FALSE FALSE      4 TRUE               7       20  0           0.496
5 AADAT  FALSE FALSE      4 TRUE              16       32  0.0000330  -0.366
6 AAK1   FALSE FALSE      5 TRUE             553     1324  0.00000646  0.560
#  2 more variables: adj.P.Val &lt;dbl&gt;, module_label &lt;chr&gt;</code></pre>
</div>
</div>
<p><strong>Save Results</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>  key_modules,</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">file.path</span>(</span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>    output_dir,</span>
<span id="cb141-5"><a href="#cb141-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(tag, <span class="st">"_key_modules_ranked.csv"</span>)</span>
<span id="cb141-6"><a href="#cb141-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb141-7"><a href="#cb141-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extraction-of-key-differentially-expressed-genes" class="level2">
<h2 class="anchored" data-anchor-id="extraction-of-key-differentially-expressed-genes">Extraction of Key Differentially Expressed Genes</h2>
<p>From the annotated network, we extract the top 20 upregulated and top 20 downregulated genes. These represent highly differentiated genes in the Tumor vs Normal contrast and are used for summarizing key biological processes in the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>top20_up <span class="ot">&lt;-</span> nodes_out <span class="sc">|&gt;</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(logFC),</span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a>    logFC <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(adj.P.Val),</span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a>    adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb142-7"><a href="#cb142-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb142-8"><a href="#cb142-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(logFC)) <span class="sc">|&gt;</span></span>
<span id="cb142-9"><a href="#cb142-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">|&gt;</span></span>
<span id="cb142-10"><a href="#cb142-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="st">"Upregulated"</span>)</span>
<span id="cb142-11"><a href="#cb142-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-12"><a href="#cb142-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top20_up)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  12
  symbol hub   seed  module in_key_module degree strength  betweenness logFC
  &lt;chr&gt;  &lt;lgl&gt; &lt;lgl&gt;  &lt;dbl&gt; &lt;lgl&gt;          &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
1 PIGR   FALSE TRUE       1 TRUE             392      784 0.00000102    8.52
2 ADIPOQ FALSE TRUE       8 TRUE              94      189 0.0000468     8.17
3 SCARA5 FALSE TRUE       6 TRUE              30       73 0.0000000630  7.03
4 RBP4   FALSE TRUE       1 TRUE              54      179 0.00000117    7.01
5 LEP    FALSE TRUE       8 TRUE              97      216 0.0000443     6.94
6 PLIN1  FALSE TRUE       8 TRUE             190      502 0.0000756     6.42
#  3 more variables: adj.P.Val &lt;dbl&gt;, module_label &lt;chr&gt;, direction &lt;chr&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>top20_down <span class="ot">&lt;-</span> nodes_out <span class="sc">|&gt;</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(logFC),</span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a>    logFC <span class="sc">&lt;</span> <span class="dv">0</span>,</span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(adj.P.Val),</span>
<span id="cb144-6"><a href="#cb144-6" aria-hidden="true" tabindex="-1"></a>    adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb144-7"><a href="#cb144-7" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb144-8"><a href="#cb144-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(logFC) <span class="sc">|&gt;</span></span>
<span id="cb144-9"><a href="#cb144-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">|&gt;</span></span>
<span id="cb144-10"><a href="#cb144-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="st">"Downregulated"</span>)</span>
<span id="cb144-11"><a href="#cb144-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-12"><a href="#cb144-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(top20_down)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  12
  symbol  hub   seed  module in_key_module degree strength betweenness logFC
  &lt;chr&gt;   &lt;lgl&gt; &lt;lgl&gt;  &lt;dbl&gt; &lt;lgl&gt;          &lt;dbl&gt;    &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt;
1 COL11A1 FALSE TRUE       6 TRUE             220      830  0.0000190  -7.67
2 COL10A1 FALSE TRUE       6 TRUE             198      696  0.00000835 -7.34
3 CXCL10  FALSE TRUE       2 TRUE             605     1671  0.000114   -6.71
4 VCAN    FALSE TRUE       1 TRUE             347     1151  0.000206   -6.67
5 RRM2    FALSE TRUE       3 TRUE             555     1439  0.000747   -6.64
6 TOP2A   FALSE TRUE       3 TRUE             583     1589  0.000126   -6.41
#  3 more variables: adj.P.Val &lt;dbl&gt;, module_label &lt;chr&gt;, direction &lt;chr&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>combined_40 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>  top20_up,</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>  top20_down</span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(</span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">desc</span>(<span class="fu">abs</span>(logFC))</span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combined_40)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  12
  symbol  hub   seed  module in_key_module degree strength  betweenness logFC
  &lt;chr&gt;   &lt;lgl&gt; &lt;lgl&gt;  &lt;dbl&gt; &lt;lgl&gt;          &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
1 PIGR    FALSE TRUE       1 TRUE             392      784 0.00000102    8.52
2 ADIPOQ  FALSE TRUE       8 TRUE              94      189 0.0000468     8.17
3 COL11A1 FALSE TRUE       6 TRUE             220      830 0.0000190    -7.67
4 COL10A1 FALSE TRUE       6 TRUE             198      696 0.00000835   -7.34
5 SCARA5  FALSE TRUE       6 TRUE              30       73 0.0000000630  7.03
6 RBP4    FALSE TRUE       1 TRUE              54      179 0.00000117    7.01
#  3 more variables: adj.P.Val &lt;dbl&gt;, module_label &lt;chr&gt;, direction &lt;chr&gt;</code></pre>
</div>
</div>
<p><strong>Save Resilts</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>  combined_40,</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">file.path</span>(</span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>    output_dir,</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(tag, <span class="st">"_top40_DE.csv"</span>)</span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>