---
title: "PCA Analysis"
format: html
---

# PCA Analysis

## Load Libraries

```{r}
#| message: false
library(tidyverse)
library(ggbiplot)
library(devtools)
```

## Load Data

```{r}
expr_data <- read.csv("../data/GSE45827_clean_expression.csv", row.names = 1)
pheno_data <- read.csv("../data/GSE45827_clean_Phenotype_metadata.csv", row.names = 1)
```

## PCA Calculation And Plotting

**Format data for PCA**

```{r}
transp_expr <- t(expr_data) |> data.frame()

transp_expr <- transp_expr |>
  mutate(diagnosis = pheno_data$diagnosis)

transp_expr <- transp_expr |>
  mutate(subtype = pheno_data$subtype)

expr_only <- transp_expr |>  select(-subtype, -diagnosis)
```

**Perform PCA**

```{r}
pc <- prcomp(expr_only, center = TRUE, scale. = TRUE)
```

**PCA: Cancer vs Healthy**

```{r}
ggbiplot(pc, 
         obs.scale = 1,
         var.scale = 1,
         groups = transp_expr$diagnosis,
         var.axes = FALSE)
```

**PCA: By Cancer Subtype**

```{r}
ggbiplot(pc, 
         obs.scale = 1,
         var.scale = 1,
         groups = transp_expr$subtype,
         var.axes = FALSE)
  
```

## Data Analysis

PCA Plots are a tool for visualizing high dimensional data onto an orthogonal basis so that the first principal component (PC, aka the "axis") is aligned with the largest source of variance, and then the second PC is aligned with the second largest source of variance etc. This is used to visualize and identify underlying patterns within the data.

The second plot shows us which samples are similar in terms of their gene expression, and which ones are different, which reveals clusters of similarly behaving gene expression. As the coloring shows, the tumour subtypes are plotted in certain area of the graph. This means, there are patterns of gene expression within one subtype that differ generally from other subtypes. This proves clinic significance of the subtypes and enforces further analysis of the gene expression behavior.

The first graph shows the more general plot and instead of plotting for all the different subtypes of tumor, we are looking at the genes that are highly expressed in all the breast cancers vs in normal cells, and colored based on these two variables.
