---
title: "Differential Gene Expression Analysis"
format: 
  html:
    self-contained: true
---

# Differential Gene Expression Analysis

## Load Libraries

```{r}
#| message: false
library(limma)
library(dplyr)
library(msigdbr)
library(fgsea)
library(DESeq2)
library(purrr)
library(ggplot2)
```

## Load Data

```{r}
expr_data <- read.csv(
  "../data/GSE45827_clean_expression.csv",
  row.names = 1
)

feature_data <- read.csv(
  "../data/GSE45827_clean_Feature_metadata.csv",
  row.names = 1
)

pheno_data <- read.csv(
  "../data/GSE45827_clean_Phenotype_metadata.csv",
  row.names = 1
)
```

## Linear Modeling and Differential Expression Analysis

```{r}

groups <- as.factor(pheno_data |> pull(subtype))
design <- model.matrix(~ 0 + groups)

clean_levels <- make.names(levels(groups))
colnames(design) <- clean_levels

fit <- lmFit(expr_data, design)

contrast_matrix <- makeContrasts(
  Basal_vs_Her2 = Basal - Her2,
  Basal_vs_LumA = Basal - Luminal.A,
  Her2_vs_LumA = Her2 - Luminal.A,
  LumA_vs_LumB = Luminal.A - Luminal.B,
  levels = design
)

fit_contrast <- contrasts.fit(fit, contrast_matrix)
fit_ebayes <- eBayes(fit_contrast)

summary(decideTests(fit_ebayes))
```

## Preparing the Gene List

```{r}
top_table <- topTable(fit_ebayes, coef = "Basal_vs_LumA", number = Inf)

head(top_table)

top_table <- top_table |> 
  rownames_to_column(var = "probe_id")

probe_to_gene_map <- data.frame(
  probe_id = rownames(feature_data),
  gene_symbol = feature_data |> pull("Gene.Symbol") )

res_with_symbols <- merge(top_table, probe_to_gene_map, by="probe_id")

res_with_symbols <- res_with_symbols |> 
  filter(!is.na(t) & !is.na(gene_symbol) & gene_symbol != "")

res_ranked <- res_with_symbols |> 
  group_by(gene_symbol)  |> 
  slice_max(order_by = abs(t), n = 1) |> 
  ungroup()

stats <- res_ranked$t
names(stats) <- res_ranked |> pull(gene_symbol)

stats <- sort(stats, decreasing = TRUE)

head(stats)
```

## Running GSEA

```{r}
BP_df <- msigdbr(species = "human", category = "C5", subcategory = "BP")

pathways <- split(BP_df |> pull(gene_symbol), f = BP_df |> pull(gs_name))

fgseaRes <- fgseaMultilevel(
  pathways = pathways,
  stats    = stats,
  minSize  = 15,
  maxSize  = 500
)

print("Top GSEA results (Basal_vs_LumA):")
head(fgseaRes[order(padj), ])
```

## The Enrichment Plot

```{r}
fgseaRes_ordered <- fgseaRes |> 
  arrange(padj)

top_pathway <- fgseaRes_ordere |> 
  slice(1) |> 
  select(pathway, leadingEdge)

top_pathway_name <- top_pathway |> pull(pathway)
leading_edge_genes <- top_pathway |> pull(leadingEdge)  |>  .[[1]]

first_genes <- leading_edge_genes |> head(10)
print(first_genes)

plotEnrichment(pathways[[top_pathway_name]], stats = stats) +
  labs(title = top_pathway_name)
```

## Getting the Top Genes

```{r}

top_5 <- fgseaRes_ordered |> 
  slice(1:5) |> 
  select(pathway, leadingEdge)

top_5_names <- top_5 |> pull(pathway)
top_5_gene_lists <- top_5 |> pull(leadingEdge)

walk(top_5_gene_lists, ~ print(head(.x, 10)))
```

## Intersection Analysis of Core Enrichment Genes

```{r}
core_drivers <- reduce(top_5_gene_lists, intersect)

res <- top_5_gene_lists |> 
  flatten_chr() |> 
  as_tibble()  |> 
  count(value, sort = TRUE) |> 
  filter(n >= 3)

res
```

## Saving Results

```{r}
fgseaRes_flat <- fgseaRes

fgseaRes_flat$leadingEdge <- sapply(fgseaRes_flat$leadingEdge, paste, collapse = "; ")

write.csv(fgseaRes_flat, "../data/Top_GSEA_results_Basal_vs_LumA.csv", row.names = FALSE)

top_5_gene_df <- map_dfr(top_5_gene_lists, ~as.data.frame(t(.x), .id="id"))
write.csv(top_5_gene_df, "../data/Genes_from_Top_5_pathways.csv")
write.csv(res, "../data/Intersected_Genes.csv", row.names = FALSE)

my_plot <- plotEnrichment(pathways[[top_pathway_name]], stats = stats) +
           labs(title = top_pathway_name)

ggsave(filename = "../results/Intersected_Genes_Plot.png", 
       plot = my_plot, 
       width = 8, 
       height = 6)
```
